From 3827ded8af944d88d1a0d9c367a4b7c84bbd5650 Mon Sep 17 00:00:00 2001
From: martinRenou <martin.renou@gmail.com>
Date: Thu, 11 Jan 2024 10:51:23 +0100
Subject: [PATCH 1/1] Don't run find_package(PythonInterp)

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 28450f4..a0106a2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,7 +39,7 @@ endif ()
 # Running find_package(PythonInterp) to retrieve the Python version
 # which is not exported by Pybind11's cmake.
 # Cf. https://github.com/pybind/pybind11/issues/2268
-find_package(PythonInterp ${PythonLibsNew_FIND_VERSION} REQUIRED)
+# find_package(PythonInterp ${PythonLibsNew_FIND_VERSION} REQUIRED)
 
 configure_file (
     "${CMAKE_CURRENT_SOURCE_DIR}/share/jupyter/kernels/xpython/kernel.json.in"
-- 
2.43.0

