From 80f01a8dd1529a45b1cd37d59e6fabddcc68582c Mon Sep 17 00:00:00 2001
From: Isabel Paredes <isabel.paredes@quantstack.net>
Date: Wed, 3 Jun 2026 17:49:18 +0200
Subject: [PATCH] Prevent double linking flang_rt.runtime

---
 src/Makevars | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makevars b/src/Makevars
index b37b1b4..16caf3f 100644
--- a/src/Makevars
+++ b/src/Makevars
@@ -1,2 +1,2 @@
 ## we use the BLAS and now also the LAPACK library:
-PKG_LIBS= $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS)
+PKG_LIBS= $(LAPACK_LIBS) $(BLAS_LIBS)
