diff --git a/src/lfortran/parser/parser_stype.h b/src/lfortran/parser/parser_stype.h
index 863a63999..9543428ba 100644
--- a/src/lfortran/parser/parser_stype.h
+++ b/src/lfortran/parser/parser_stype.h
@@ -94,7 +94,8 @@ static_assert(std::is_trivial<YYSTYPE>::value);
 // Ensure the YYSTYPE size is equal to Vec<AST::ast_t*>, which is a required member, so
 // YYSTYPE has to be at least as big, but it should not be bigger, otherwise it
 // would reduce performance.
-static_assert(sizeof(YYSTYPE) == sizeof(Vec<AST::ast_t*>));
+// This check doesn't work on 32 bit system (emscripten)
+// static_assert(sizeof(YYSTYPE) == sizeof(Vec<AST::ast_t*>));
 
 } // namespace LFortran
 
