Submodule subprojects/highs contains modified content
diff --git a/subprojects/highs/meson.build b/subprojects/highs/meson.build
index 8cb61f8b7..bae6f0518 100644
--- a/subprojects/highs/meson.build
+++ b/subprojects/highs/meson.build
@@ -110,7 +110,7 @@ if is_mingw
 endif

 # --------------------- Dependencies
-threads_dep = dependency('threads', required: false)
+threads_dep = dependency('threads_DONTUSE', required: false)
 _deps += threads_dep

 # Determine whether it is necessary to link libatomic. This could be the case