diff --git a/LAPACKE/include/CMakeLists.txt b/LAPACKE/include/CMakeLists.txt
index 034491e..1ecf561 100644
--- a/LAPACKE/include/CMakeLists.txt
+++ b/LAPACKE/include/CMakeLists.txt
@@ -14,7 +14,7 @@ if(CMAKE_Fortran_COMPILER AND CMAKE_C_COMPILER)
   enable_language(C)
   include(FortranCInterface)
   ## Ensure that the fortran compiler and c compiler specified are compatible
-  FortranCInterface_VERIFY()
+  # FortranCInterface_VERIFY()
   FortranCInterface_HEADER(${LAPACK_BINARY_DIR}/include/lapacke_mangling.h
     MACRO_NAMESPACE "LAPACK_"
     SYMBOL_NAMESPACE "LAPACK_")
