diff --git a/BLAS/SRC/CMakeLists.txt b/BLAS/SRC/CMakeLists.txt
index 9df128e..2f4d478 100644
--- a/BLAS/SRC/CMakeLists.txt
+++ b/BLAS/SRC/CMakeLists.txt
@@ -126,7 +126,7 @@ if(BUILD_INDEX64_EXT_API)
     if(CMAKE_Fortran_COMPILER_ID STREQUAL "NAG")
       set(COPT_64_F -fpp)
     else()
-      set(COPT_64_F -cpp)
+      set(COPT_64_F --cpp)
     endif()
     file(STRINGS ${F} ${F}.lst)
     list(FILTER ${F}.lst INCLUDE REGEX "subroutine|SUBROUTINE|external|EXTERNAL|function|FUNCTION")
