diff --git a/cmake/OpenEXRSetup.cmake b/cmake/OpenEXRSetup.cmake
index 898847f..d27ba1a 100644
--- a/cmake/OpenEXRSetup.cmake
+++ b/cmake/OpenEXRSetup.cmake
@@ -405,7 +405,7 @@ endif()
 # Enable SSE2 on 32-bit x86
 ###########################################
 
-if(CMAKE_SYSTEM_PROCESSOR MATCHES "^(i[3-6]86|x86)$" AND NOT MSVC)
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "^(i[3-6]86|x86)$" AND NOT MSVC AND NOT EMSCRIPTEN)
   include(CheckCCompilerFlag)
   check_c_compiler_flag("-msse2" HAS_SSE2_FLAG)
   if(HAS_SSE2_FLAG)
