diff --git a/clang/lib/Driver/Driver.cpp b/clang/lib/Driver/Driver.cpp
--- a/clang/lib/Driver/Driver.cpp
+++ b/clang/lib/Driver/Driver.cpp
@@ -2617,7 +2617,7 @@
       C.getArgs().hasArg(options::OPT_print_supported_cpus) ||
       C.getArgs().hasArg(options::OPT_print_supported_extensions) ||
       C.getArgs().hasArg(options::OPT_print_enabled_extensions)) {
-    PrintVersion(C, llvm::errs());
+    PrintVersion(C, llvm::outs());
     SuppressMissingInputWarning = true;
   }
 
@@ -2633,7 +2633,7 @@
   const ToolChain &TC = C.getDefaultToolChain();
 
   if (C.getArgs().hasArg(options::OPT_v))
-    TC.printVerboseInfo(llvm::errs());
+    TC.printVerboseInfo(llvm::outs());
 
   if (C.getArgs().hasArg(options::OPT_print_resource_dir)) {
     llvm::outs() << ResourceDir << '\n';
diff --git a/clang/lib/Interpreter/Interpreter.cpp b/clang/lib/Interpreter/Interpreter.cpp
--- a/clang/lib/Interpreter/Interpreter.cpp
+++ b/clang/lib/Interpreter/Interpreter.cpp
@@ -267,7 +267,7 @@
       return std::move(Err);
 
   if (Compilation->getArgs().hasArg(options::OPT_v))
-    Compilation->getJobs().Print(llvm::errs(), "\n", /*Quote=*/false);
+    Compilation->getJobs().Print(llvm::outs(), "\n", /*Quote=*/false);
 
   auto ErrOrCC1Args = GetCC1Arguments(&Diags, Compilation.get());
   if (auto Err = ErrOrCC1Args.takeError())
@@ -337,6 +337,7 @@
                          std::unique_ptr<clang::ASTConsumer> Consumer)
     : IncrExecutorBuilder(std::move(IEB)) {
   CI = std::move(Instance);
+  CI->setVerboseOutputStream(llvm::outs());
   llvm::ErrorAsOutParameter EAO(&ErrOut);
   auto LLVMCtx = std::make_unique<llvm::LLVMContext>();
   TSCtx = std::make_unique<llvm::orc::ThreadSafeContext>(std::move(LLVMCtx));
diff --git a/clang/lib/Lex/InitHeaderSearch.cpp b/clang/lib/Lex/InitHeaderSearch.cpp
--- a/clang/lib/Lex/InitHeaderSearch.cpp
+++ b/clang/lib/Lex/InitHeaderSearch.cpp
@@ -165,7 +165,7 @@
   }
 
   if (Verbose)
-    llvm::errs() << "ignoring nonexistent directory \""
+    llvm::outs() << "ignoring nonexistent directory \""
                  << MappedPathStr << "\"\n";
   return false;
 }
@@ -347,10 +347,10 @@
     }
 
     if (Verbose) {
-      llvm::errs() << "ignoring duplicate directory \""
+      llvm::outs() << "ignoring duplicate directory \""
                    << CurEntry.getName() << "\"\n";
       if (DirToRemove != i)
-        llvm::errs() << "  as it is a non-system directory that duplicates "
+        llvm::outs() << "  as it is a non-system directory that duplicates "
                      << "a system directory\n";
     }
     if (DirToRemove != i)
@@ -447,9 +447,9 @@
         assert(SearchList[i].Lookup.isHeaderMap() && "Unknown DirectoryLookup");
         Suffix = " (headermap)";
       }
-      llvm::errs() << " " << Name << Suffix << "\n";
+      llvm::outs() << " " << Name << Suffix << "\n";
     }
-    llvm::errs() << "End of search list.\n";
+    llvm::outs() << "End of search list.\n";
   }
 }
 
