diff --git a/clang/lib/Interpreter/Wasm.cpp b/clang/lib/Interpreter/Wasm.cpp
index aa10b160ccf8..169039e4ced8 100644
--- a/clang/lib/Interpreter/Wasm.cpp
+++ b/clang/lib/Interpreter/Wasm.cpp
@@ -55,6 +55,7 @@ bool link(llvm::ArrayRef<const char *> args, llvm::raw_ostream &stdoutOS,
 } // namespace lld
 
 #include <dlfcn.h>
+#include <unistd.h>
 
 namespace clang {
 
@@ -65,6 +66,19 @@ WasmIncrementalExecutor::WasmIncrementalExecutor(
 llvm::Error WasmIncrementalExecutor::addModule(PartialTranslationUnit &PTU) {
   std::string ErrorString;
 
+  char currentDir[256];
+  if (getcwd(currentDir, sizeof(currentDir)) == nullptr) {
+    return llvm::make_error<llvm::StringError>("Failed to get current directory",
+                                               llvm::inconvertibleErrorCode());
+  }
+
+  if (std::strcmp(currentDir, "/tmp") != 0) {
+    if (chdir("/tmp") != 0) {
+      return llvm::make_error<llvm::StringError>("Failed to change directory to /tmp",
+                                                 llvm::inconvertibleErrorCode());
+    }
+  }
+
   const llvm::Target *Target = llvm::TargetRegistry::lookupTarget(
       PTU.TheModule->getTargetTriple(), ErrorString);
   if (!Target) {
