--- cffi_example/build_fnmatch.py	2022-04-14 10:29:26.201441966 +0200
+++ cffi_example/build_fnmatch.py	2022-04-14 10:27:06.974301875 +0200
@@ -17,8 +17,8 @@
     #    }
     "#include <fnmatch.h>",
     # The important thing is to inclue libc in the list of libraries we're
-    # linking against:
-    libraries=["c"],
+    # linking against: IN EMSCRIPTEN WE DO NOT WANT THIS
+    # libraries=["c"],
 )
 
 with open(os.path.join(os.path.dirname(__file__), "fnmatch.h")) as f:
