From 7355d92d5ade339ef4de076bc3d8a5b7772bed4f Mon Sep 17 00:00:00 2001
From: Isabel Paredes <isabel.paredes@quantstack.net>
Date: Wed, 21 Aug 2024 12:42:11 +0200
Subject: [PATCH 3/9] Remove unsupported sigsuspend

---
 src/unix/sys-unix.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/unix/sys-unix.c b/src/unix/sys-unix.c
index c20afe6..f6ac6a9 100644
--- a/src/unix/sys-unix.c
+++ b/src/unix/sys-unix.c
@@ -480,7 +480,8 @@ static pid_t timeout_wait(int *wstatus)
 
     int saveerrno = errno;
     while((wres = waitpid(tost.child_pid, wstatus, WNOHANG)) == 0)
-	sigsuspend(&unblocked_ss);
+	// sigsuspend(&unblocked_ss); Not implemented in Emscripten
+	errno = EINTR;
 
     if (errno == EINTR)
 	/* EINTR is not really an error but expected situation here, however,
