From 9b11ddff35b56c1b2b5141a37b3a790e4cd662b7 Mon Sep 17 00:00:00 2001
From: Isabel Paredes <isabel.paredes@quantstack.net>
Date: Wed, 21 Aug 2024 13:51:47 +0200
Subject: [PATCH 8/9] Set default R_Home

---
 src/unix/system.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/unix/system.c b/src/unix/system.c
index ff5751d..84cf79c 100644
--- a/src/unix/system.c
+++ b/src/unix/system.c
@@ -329,8 +329,8 @@ int Rf_initialize_R(int ac, char **av)
 
     R_GlobalContext = NULL; /* Make R_Suicide less messy... */
 
-    if((R_Home = R_HomeDir()) == NULL)
-	R_Suicide("R home directory is not defined");
+    if((R_Home = R_HomeDir()) == NULL) R_Home = "lib/R";
+	// R_Suicide("R home directory is not defined");
     BindDomain(R_Home);
 
     process_system_Renviron();
