# Fixes:
# ./configure: line 24267: ${}: bad substitution

diff --git a/configure b/configure
index ea413eb..4eaba16 100755
--- a/configure
+++ b/configure
@@ -24264,8 +24264,8 @@ case "${host_os}" in
 esac
 
 ## Export LD_LIBRARY_PATH or equivalent.
-if eval "test -z \"\${${Rshlibpath_var}}\""; then
-  eval "${Rshlibpath_var}=\"${R_LD_LIBRARY_PATH}\""
+if test -z ${Rshlibpath_var}; then
+  export Rshlibpath_var=${R_LD_LIBRARY_PATH}
 else
   eval "${Rshlibpath_var}=\"${R_LD_LIBRARY_PATH}${PATH_SEPARATOR}\${${Rshlibpath_var}}\""
 fi
