diff --git a/configure b/configure
index 08f67cb..a21f317 100755
--- a/configure
+++ b/configure
@@ -49611,6 +49611,7 @@ if test "$want_mbcs_support" = yes ; then
   ac_fn_c_check_type "$LINENO" "wctrans_t" "ac_cv_type_wctrans_t" "#include <wchar.h>
        #include <wctype.h>
 "
+ac_cv_type_wctrans_t=yes
 if test "x$ac_cv_type_wctrans_t" = xyes
 then :
 
@@ -49621,6 +49622,7 @@ fi
 ac_fn_c_check_type "$LINENO" "mbstate_t" "ac_cv_type_mbstate_t" "#include <wchar.h>
        #include <wctype.h>
 "
+ac_cv_type_mbstate_t=yes
 if test "x$ac_cv_type_mbstate_t" = xyes
 then :
 
@@ -49637,7 +49639,7 @@ fi
   fi
 fi
 if test "x${want_mbcs_support}" != xyes; then
-as_fn_error $? "Support for MBCS locales is required." "$LINENO" 5
+as_fn_set_status $? "Support for MBCS locales is required." "$LINENO" 5
 fi
 
 
