diff --git a/configure b/configure
index b94a657..b118e78 100755
--- a/configure
+++ b/configure
@@ -27550,7 +27550,8 @@ case "${ac_cv_fc_mangling}" in
   "upper "*)
     { printf "%s\n" "$as_me:${as_lineno-$LINENO}: WARNING: Fortran compiler uses uppercase external names" >&5
 printf "%s\n" "$as_me: WARNING: Fortran compiler uses uppercase external names" >&2;}
-    as_fn_error $? "cannot use Fortran" "$LINENO" 5
+    # as_fn_error $? "cannot use Fortran" "$LINENO" 5
+    echo "WARNING: cannot use Fortran"
     ;;
 esac
 { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking whether ${FC} appends underscores to external names" >&5
@@ -27575,7 +27576,8 @@ printf "%s\n" "${r_cv_prog_fc_append_underscore}" >&6; }
 else
   { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: unknown" >&5
 printf "%s\n" "unknown" >&6; }
-  as_fn_error $? "cannot use Fortran" "$LINENO" 5
+  # as_fn_error $? "cannot use Fortran" "$LINENO" 5
+  echo "WARNING: cannot use Fortran"
 fi
 if test "${r_cv_prog_fc_append_underscore}" = yes; then
 
@@ -27604,7 +27606,8 @@ printf "%s\n" "${r_cv_prog_fc_append_second_underscore}" >&6; }
 else
   { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: unknown" >&5
 printf "%s\n" "unknown" >&6; }
-  as_fn_error $? "cannot use Fortran" "$LINENO" 5
+  # as_fn_error $? "cannot use Fortran" "$LINENO" 5
+  echo "WARNING: cannot use Fortran"
 fi
 if test "${r_cv_prog_fc_append_second_underscore}" = yes; then
 
