diff --git a/src/Makefile.in b/src/Makefile.in
index 2666475..bcb5fe2 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -14,7 +14,7 @@ distdir = $(top_builddir)/$(PACKAGE)-$(VERSION)/$(subdir)
 DISTFILES = Makefile.in
 
 ## need extra early to get libintl.h installed
-SUBDIRS = scripts include extra appl nmath unix main modules library
+SUBDIRS = scripts include extra appl nmath main modules library
 SUBDIRS_WITH_NO_BUILD = gnuwin32
 
 all: Makefile R
