From f5ad0d8afe9ee2f5b7b7a3f39f805e9df73e7fe4 Mon Sep 17 00:00:00 2001
From: Isabel Paredes <isabel.paredes@quantstack.net>
Date: Tue, 30 Jul 2024 15:12:13 +0200
Subject: [PATCH] Skip bad substitution

---
 etc/ldpaths.in | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/etc/ldpaths.in b/etc/ldpaths.in
index 314d364..ac201a8 100644
--- a/etc/ldpaths.in
+++ b/etc/ldpaths.in
@@ -13,9 +13,9 @@ fi
 ## However, on macOS >=10.11 (if SIP is enabled, the default), the
 ## environment value will not be passed to a script such as R.sh, so
 ## would not seen here.
-if test -z "${@Rshlibpath_var@}"; then
-  @Rshlibpath_var@="${R_LD_LIBRARY_PATH}"
-else
-  @Rshlibpath_var@="${R_LD_LIBRARY_PATH}:${@Rshlibpath_var@}"
-fi
-export @Rshlibpath_var@
+# if test -z "${@Rshlibpath_var@}"; then
+#   @Rshlibpath_var@="${R_LD_LIBRARY_PATH}"
+# else
+#   @Rshlibpath_var@="${R_LD_LIBRARY_PATH}:${@Rshlibpath_var@}"
+# fi
+# export @Rshlibpath_var@
