From 4896194900006ed24cb8e6474d7d5124f613a07c Mon Sep 17 00:00:00 2001
From: Isabel Paredes <isabel.paredes@quantstack.net>
Date: Tue, 6 Aug 2024 18:37:07 +0200
Subject: [PATCH] Skip .m.d

---
 configure | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/configure b/configure
index a21f317..599ee46 100644
--- a/configure
+++ b/configure
@@ -28661,8 +28661,7 @@ EOF
 if test -n "${r_cv_prog_objc_m}"; then
   cat << EOF >> ${r_objc_rules_frag}
 .m.d:
-	@echo "making \$@ from \$<"
-	@${r_cv_prog_objc_m} \$(ALL_CPPFLAGS) $< > \$@
+	@echo "NOT making \$@ from \$<"
 EOF
 else
   cat << \EOF >> ${r_cc_rules_frag}
