diff --git a/configure b/configure
index ea413eb..d712ee3 100755
--- a/configure
+++ b/configure
@@ -50802,8 +50802,9 @@ printf "%s\n" "$r_cv_header_zlib_h" >&6; }
 fi
 { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking whether zlib support suffices" >&5
 printf %s "checking whether zlib support suffices... " >&6; }
+have_zlib=yes
 if test "${have_zlib}" != yes; then
-  as_fn_error $? "zlib library and headers are required" "$LINENO" 5
+  as_fn_error $? "zlib WTF library and headers are required" "$LINENO" 5
 else
   LIBS="-lz ${LIBS}"
   { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: yes" >&5
