From f910e84848f5e0fcf97bd793c2fcfbbc521dd704 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio.traversaro@iit.it>
Date: Wed, 20 Jul 2022 09:36:44 +0200
Subject: [PATCH] Use CONFIG for LibXml2

---
 cmake/iDynTreeDependencies.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/iDynTreeDependencies.cmake b/cmake/iDynTreeDependencies.cmake
index 20db108711f..f08a978dc27 100644
--- a/cmake/iDynTreeDependencies.cmake
+++ b/cmake/iDynTreeDependencies.cmake
@@ -63,7 +63,7 @@ if(NOT TARGET Eigen3::Eigen)
 endif()
 
 if(NOT TARGET LibXml2::LibXml2)
-  find_package(LibXml2 REQUIRED)
+  find_package(LibXml2 REQUIRED CONFIG)
 endif()
 
 idyntree_handle_dependency(YARP COMPONENTS os dev math rosmsg idl_tools MAIN_TARGET YARP::YARP_os)
