From 4b15585fef39d5a662056d7ac522a9b29a957498 Mon Sep 17 00:00:00 2001
From: Ian Thomas <ianthomas23@gmail.com>
Date: Mon, 2 Dec 2024 11:28:39 +0000
Subject: [PATCH] Use const char* args to OS term functions

---
 src/osdef2.h.in | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/osdef2.h.in b/src/osdef2.h.in
index f29451ddf..88d9af20d 100644
--- a/src/osdef2.h.in
+++ b/src/osdef2.h.in
@@ -57,11 +57,11 @@ extern int	link(const char *, const char *);
 extern int	mkdir(const char *, mode_t);
 extern int	rmdir(const char *);

-extern int	tgetent(char *, char *);
-extern int	tgetnum(char *);
-extern int	tgetflag(char *);
-extern char	*tgoto(char *, int, int);
-extern int	tputs(char *, int, int (*)(int));
+extern int	tgetent(char *, const char *);
+extern int	tgetnum(const char *);
+extern int	tgetflag(const char *);
+extern char	*tgoto(const char *, int, int);
+extern int	tputs(const char *, int, int (*)(int));

 #ifdef HAVE_TERMIOS_H
 struct termios;		/* for tcgetattr() */
--
2.43.0

