From 189c7241b9de072178ae752214d4bbae2fa674e7 Mon Sep 17 00:00:00 2001
From: Ian Thomas <ianthomas23@gmail.com>
Date: Fri, 9 Jan 2026 10:09:38 +0000
Subject: [PATCH 02/15] Remove duplicate symbol

---
 .../linalg/_dsolve/SuperLU/SRC/dcomplex.c     | 40 +++++++++----------
 1 file changed, 20 insertions(+), 20 deletions(-)

diff --git a/scipy/sparse/linalg/_dsolve/SuperLU/SRC/dcomplex.c b/scipy/sparse/linalg/_dsolve/SuperLU/SRC/dcomplex.c
index 0da48a9428..bd4f74ad49 100644
--- a/scipy/sparse/linalg/_dsolve/SuperLU/SRC/dcomplex.c
+++ b/scipy/sparse/linalg/_dsolve/SuperLU/SRC/dcomplex.c
@@ -57,27 +57,27 @@ void z_div(doublecomplex *c, doublecomplex *a, doublecomplex *b)
     c->i = ci;
 }
 
-
-/*! \brief Returns sqrt(z.r^2 + z.i^2) */
-double z_abs(doublecomplex *z)
-{
-    double temp;
-    double real = z->r;
-    double imag = z->i;
-
-    if (real < 0) real = -real;
-    if (imag < 0) imag = -imag;
-    if (imag > real) {
-	temp = real;
-	real = imag;
-	imag = temp;
-    }
-    if ((real+imag) == real) return(real);
+// NOTE: Already defined in openblas
+// /*! \brief Returns sqrt(z.r^2 + z.i^2) */
+// double z_abs(doublecomplex *z)
+// {
+//     double temp;
+//     double real = z->r;
+//     double imag = z->i;
+
+//     if (real < 0) real = -real;
+//     if (imag < 0) imag = -imag;
+//     if (imag > real) {
+// 	temp = real;
+// 	real = imag;
+// 	imag = temp;
+//     }
+//     if ((real+imag) == real) return(real);
   
-    temp = imag/real;
-    temp = real*sqrt(1.0 + temp*temp);  /*overflow!!*/
-    return (temp);
-}
+//     temp = imag/real;
+//     temp = real*sqrt(1.0 + temp*temp);  /*overflow!!*/
+//     return (temp);
+// }
 
 
 /*! \brief Approximates the abs. Returns abs(z.r) + abs(z.i) */
-- 
2.51.0

