--- scipy/_lib/array_api_compat/array_api_compat/_internal.py	2026-02-20 10:57:07.220973331 +0000
+++ scipy/_lib/array_api_compat/array_api_compat/_internal.py	2026-02-20 10:57:47.303833856 +0000
@@ -61,10 +61,10 @@
     # Neither of these two methods is sufficient by itself,
     # depending on various idiosyncrasies of the libraries we're wrapping.
     objs = {}
-    exec(f"from {mod.__name__} import *", objs)
+    #exec(f"from {mod.__name__} import *", objs)

     for n in dir(mod):
-        if not n.startswith("_") and hasattr(mod, n):
+        if n == "__version__" or (not n.startswith("_") and n != "f2py"):
             objs[n] = getattr(mod, n)

     globals_.update(objs)
