From 19a4376b504b45813f6a2d5c12ce613625ee9c36 Mon Sep 17 00:00:00 2001
From: Ian Thomas <ianthomas23@gmail.com>
Date: Sat, 10 Jan 2026 12:25:43 +0000
Subject: [PATCH 03/18] Disable unsupported -ffloat-store

---
 scipy/optimize/_highspy/meson.build | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/scipy/optimize/_highspy/meson.build b/scipy/optimize/_highspy/meson.build
index 3b3484243d..c5e9f92ecc 100644
--- a/scipy/optimize/_highspy/meson.build
+++ b/scipy/optimize/_highspy/meson.build
@@ -26,11 +26,11 @@ highs_incdirs = highs_proj.get_variable('_incdirs')
 highs_cpp_suppressions = highs_proj.get_variable('_highs_cpp_suppressions')
 highs_c_suppressions = highs_proj.get_variable('_highs_c_suppressions')
 
-if is_32bit
-  _float_store = cc.get_supported_arguments('-ffloat-store')
-  _cpp_float_store = cpp.get_supported_arguments('-ffloat-store')
-  highs_args += _float_store + _cpp_float_store
-endif
+#if is_32bit
+#  _float_store = cc.get_supported_arguments('-ffloat-store')
+#  _cpp_float_store = cpp.get_supported_arguments('-ffloat-store')
+#  highs_args += _float_store + _cpp_float_store
+#endif
 
 highspy_cpp = highs_proj.get_variable('highspy_cpp')
 highs_conf_file = highs_proj.get_variable('highs_conf_file')
-- 
2.51.0

