From ec5a9e010e4ee0db3ca40bd43dc1b382014a5070 Mon Sep 17 00:00:00 2001
From: Ian Thomas <ianthomas23@gmail.com>
Date: Sun, 11 Jan 2026 12:27:47 +0000
Subject: [PATCH 04/18] int size in dierckxmodule

---
 scipy/interpolate/src/_dierckxmodule.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scipy/interpolate/src/_dierckxmodule.cc b/scipy/interpolate/src/_dierckxmodule.cc
index 32ad5e0c85..4be2d59c7c 100644
--- a/scipy/interpolate/src/_dierckxmodule.cc
+++ b/scipy/interpolate/src/_dierckxmodule.cc
@@ -203,7 +203,7 @@ py_fpbacp(PyObject* self, PyObject *args)
     Py_ssize_t m = PyArray_DIM(a_x, 0);
     Py_ssize_t len_t = PyArray_DIM(a_t, 0);
 
-    int64_t nc = len_t - k - 1;
+    Py_ssize_t nc = len_t - k - 1;
 
     // allocate the output buffer
     npy_intp dims[2] = {nc, PyArray_DIM(a_y, 1)};
-- 
2.51.0

