From 23b7bebe8ddf10e87b2f582a84c4840b05d6209c Mon Sep 17 00:00:00 2001
From: Isabel Paredes <isabel.paredes@quantstack.net>
Date: Thu, 13 Nov 2025 13:09:41 +0100
Subject: [PATCH 1/2] Remove unsupported version script

---
 meson.build | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/meson.build b/meson.build
index 94ffaeb..12c61f0 100644
--- a/meson.build
+++ b/meson.build
@@ -133,13 +133,14 @@ add_project_arguments(_intel_fflags, language: 'fortran')
 # Meson using `-fvisibility=hidden` for C and `-fvisibility-inlines-hidden` for
 # C++ code. See gh-15996 for details.
 _linker_script = meson.project_source_root() / 'scipy/_build_utils/link-version-pyinit.map'
-version_link_args = ['-Wl,--version-script=' + _linker_script]
-# Note that FreeBSD only accepts version scripts when -shared is passed,
-# hence we need to pass that to `cc.links` explicitly (flag is already
-# present for `extension_module` invocations).
-if not cc.links('', name: '-Wl,--version-script', args: ['-shared', version_link_args])
-  version_link_args = []
-endif
+# version_link_args = ['-Wl,--version-script=' + _linker_script]
+# # Note that FreeBSD only accepts version scripts when -shared is passed,
+# # hence we need to pass that to `cc.links` explicitly (flag is already
+# # present for `extension_module` invocations).
+# if not cc.links('', name: '-Wl,--version-script', args: ['-shared', version_link_args])
+#   version_link_args = []
+# endif
+version_link_args = []

 generate_f2pymod = find_program('tools/generate_f2pymod.py')
 tempita = find_program('scipy/_build_utils/tempita.py')
 