From 3070e2332c7d35be971a87790fd8e65ae55976ee Mon Sep 17 00:00:00 2001
From: Ian Thomas <ianthomas23@gmail.com>
Date: Fri, 9 Jan 2026 10:21:48 +0000
Subject: [PATCH 05/15] Remove unsupported version script

---
 meson.build | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/meson.build b/meson.build
index 11eb036164..4100003bd3 100644
--- a/meson.build
+++ b/meson.build
@@ -141,13 +141,14 @@ add_global_arguments(_intel_fflags, language: 'fortran')
 # Meson using `-fvisibility=hidden` for C and `-fvisibility-inlines-hidden` for
 # C++ code. See gh-15996 for details.
 _linker_script = meson.project_source_root() / 'scipy/_build_utils/link-version-pyinit.map'
-version_link_args = ['-Wl,--version-script=' + _linker_script]
-# Note that FreeBSD only accepts version scripts when -shared is passed,
-# hence we need to pass that to `cc.links` explicitly (flag is already
-# present for `extension_module` invocations).
-if not cc.links('', name: '-Wl,--version-script', args: ['-shared', version_link_args])
-  version_link_args = []
-endif
+# version_link_args = ['-Wl,--version-script=' + _linker_script]
+# # Note that FreeBSD only accepts version scripts when -shared is passed,
+# # hence we need to pass that to `cc.links` explicitly (flag is already
+# # present for `extension_module` invocations).
+# if not cc.links('', name: '-Wl,--version-script', args: ['-shared', version_link_args])
+#   version_link_args = []
+# endif
+version_link_args = []
 
 generate_f2pymod = find_program('tools/generate_f2pymod.py')
 tempita = find_program('scipy/_build_utils/tempita.py')
-- 
2.51.0

