From 65ff432dfc4e190c36ef20f055d1470d223d13f5 Mon Sep 17 00:00:00 2001
From: Ian Thomas <ianthomas23@gmail.com>
Date: Tue, 9 Dec 2025 13:09:42 +0000
Subject: [PATCH 12/15] Change arg type in odepack

---
 scipy/integrate/odepack/xerrwv.f | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/scipy/integrate/odepack/xerrwv.f b/scipy/integrate/odepack/xerrwv.f
index 7e180e4f88..141d9ac1dd 100644
--- a/scipy/integrate/odepack/xerrwv.f
+++ b/scipy/integrate/odepack/xerrwv.f
@@ -1,5 +1,6 @@
       subroutine xerrwv (msg, nmes, nerr, level, ni, i1, i2, nr, r1, r2)
-      integer msg, nmes, nerr, level, ni, i1, i2, nr,
+      character msg
+      integer nmes, nerr, level, ni, i1, i2, nr,
      1   i, lun, lunit, mesflg, ncpw, nch, nwds
       double precision r1, r2
       dimension msg(nmes)
-- 
2.51.0

