From 97ce348e61308172d0d17fa175ce55ad974e268b Mon Sep 17 00:00:00 2001
From: Ian Thomas <ianthomas23@gmail.com>
Date: Tue, 9 Dec 2025 13:19:22 +0000
Subject: [PATCH 14/15] Make sf_error_state a static library

---
 scipy/special/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scipy/special/meson.build b/scipy/special/meson.build
index 3c42247b4a..2dd6abed3b 100644
--- a/scipy/special/meson.build
+++ b/scipy/special/meson.build
@@ -33,7 +33,7 @@ else
   scipy_import_dll_args = []
 endif
 
-sf_error_state_lib = shared_library('sf_error_state',
+sf_error_state_lib = static_library('sf_error_state',
   ['sf_error_state.c'],
   include_directories: ['../_lib', '../_build_utils/src'],
   c_args: scipy_export_dll_args,
-- 
2.51.0

