diff --git a/scipy/fft/_pocketfft/setup.py b/scipy/fft/_pocketfft/setup.py
index 7e4456501..f5c682986 100644
--- a/scipy/fft/_pocketfft/setup.py
+++ b/scipy/fft/_pocketfft/setup.py
@@ -11,9 +11,9 @@ def pre_build_hook(build_ext, ext):
         args.append('/EHsc')
     else:
         # Use pthreads if available
-        has_pthreads = try_compile(cc, code='#include <pthread.h>\n'
-                                   'int main(int argc, char **argv) {}')
-        if has_pthreads:
+        # has_pthreads = try_compile(cc, code='#include <pthread.h>\n'
+        #                            'int main(int argc, char **argv) {}')
+        if False:
             ext.define_macros.append(('POCKETFFT_PTHREADS', None))
             if has_flag(cc, '-pthread'):
                 args.append('-pthread')
