diff --git a/scipy/linalg/lapack.py b/scipy/linalg/lapack.py
index 9850d80c4..09e670337 100644
--- a/scipy/linalg/lapack.py
+++ b/scipy/linalg/lapack.py
@@ -815,6 +815,11 @@ All functions
 
 import numpy as _np
 from .blas import _get_funcs, _memoize_get_funcs
+
+import ctypes
+import sys
+ctypes.CDLL(f'{sys.prefix}/lib/clapack_all.so', mode=ctypes.RTLD_GLOBAL)
+
 from scipy.linalg import _flapack
 from re import compile as regex_compile
 try:
