From cf82e885746b5adda8917c280c4be8670808e4dd Mon Sep 17 00:00:00 2001
From: Isabel Paredes <isabel.paredes@quantstack.net>
Date: Mon, 5 Jan 2026 12:07:02 +0100
Subject: [PATCH] Make tifffile optional

---
 src/skimage/io/collection.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/skimage/io/collection.py b/src/skimage/io/collection.py
index c87c5a5..be6caf1 100644
--- a/src/skimage/io/collection.py
+++ b/src/skimage/io/collection.py
@@ -9,7 +9,10 @@ from copy import copy
 import numpy as np
 from PIL import Image
 
-from tifffile import TiffFile
+try:
+    from tifffile import TiffFile
+except ImportError:
+    TiffFile = None
 
 
 __all__ = [
