diff --git a/scikitplot/nc/_linalg/src/module_linalg.cpp b/scikitplot/nc/_linalg/src/module_linalg.cpp
--- a/scikitplot/nc/_linalg/src/module_linalg.cpp
+++ b/scikitplot/nc/_linalg/src/module_linalg.cpp
@@ -23,5 +23,5 @@
-PYBIND11_MODULE(_linalg, m, py::multiple_interpreters::per_interpreter_gil()) {
+PYBIND11_MODULE(_linalg, m) {
     m.doc() = "Linear algebra core utilities";

diff --git a/scikitplot/nc/_version/src/module_version.cpp b/scikitplot/nc/_version/src/module_version.cpp
--- a/scikitplot/nc/_version/src/module_version.cpp
+++ b/scikitplot/nc/_version/src/module_version.cpp
@@ -102,5 +102,5 @@
-PYBIND11_MODULE(_version, m, py::multiple_interpreters::per_interpreter_gil()) {
+PYBIND11_MODULE(_version, m) {
     m.doc() = "Version tracking utilities";
