diff --git a/scikitplot/nc/_linalg/src/module_linalg.cpp b/scikitplot/nc/_linalg/src/module_linalg.cpp
--- a/scikitplot/nc/_linalg/src/module_linalg.cpp
+++ b/scikitplot/nc/_linalg/src/module_linalg.cpp
@@ -21,6 +21,3 @@
-    _linalg,
-    m,
-    py::mod_gil_not_used(),
-    py::multiple_interpreters::per_interpreter_gil()){
+    _linalg,
+    m){
 
diff --git a/scikitplot/nc/_version/src/module_version.cpp b/scikitplot/nc/_version/src/module_version.cpp
--- a/scikitplot/nc/_version/src/module_version.cpp
+++ b/scikitplot/nc/_version/src/module_version.cpp
@@ -21,6 +21,3 @@
-    __version,
-    m,
-    py::mod_gil_not_used(),
-    py::multiple_interpreters::per_interpreter_gil()){
+    __version,
+    m){
 
diff --git a/tools/submodules/vendored_NumCpp/src/nc.cpp b/tools/submodules/vendored_NumCpp/src/nc.cpp
--- a/tools/submodules/vendored_NumCpp/src/nc.cpp
+++ b/tools/submodules/vendored_NumCpp/src/nc.cpp
@@ -21,6 +21,3 @@
-    nc,
-    m,
-    py::mod_gil_not_used(),
-    py::multiple_interpreters::per_interpreter_gil()){
+    nc,
+    m){
 
diff --git a/scikitplot/_core/src/dummymodule.c b/scikitplot/_core/src/dummymodule.c
--- a/scikitplot/_core/src/dummymodule.c
+++ b/scikitplot/_core/src/dummymodule.c
@@ -2,9 +2,2 @@
-static struct PyModuleDef_Slot dummy_slots[] = {
-#if PY_VERSION_HEX >= 0x030c00f0  // Python 3.12+
-    // signal that this module can be imported in isolated subinterpreters
-    {Py_mod_multiple_interpreters, Py_MOD_PER_INTERPRETER_GIL_SUPPORTED},
-#endif
-#if PY_VERSION_HEX >= 0x030d00f0  // Python 3.13+
-    // signal that this module supports running without an active GIL
-#endif
+static struct PyModuleDef_Slot dummy_slots[] = {