diff --git a/meson.build b/meson.build
index f3e43b595..ed5ee6448 100644
--- a/meson.build
+++ b/meson.build
@@ -366,13 +366,6 @@ endif
 
 # threads
 thread_dep = null_dep
-if host_machine.system() != 'windows'
-  thread_dep = dependency('threads', required: false)
-
-  if thread_dep.found()
-    conf.set('HAVE_PTHREAD', 1)
-  endif
-endif
 
 conf.set_quoted('PACKAGE_NAME', 'HarfBuzz')
 conf.set_quoted('PACKAGE_VERSION', meson.project_version())
