From fa7e50c6b4f83121127fe452a7424f71f08780d7 Mon Sep 17 00:00:00 2001
From: Ian Thomas <ianthomas23@gmail.com>
Date: Mon, 15 Jun 2026 14:43:12 +0100
Subject: [PATCH] Disable threads

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 97ae4ce1b..1bf45b9cc 100644
--- a/meson.build
+++ b/meson.build
@@ -349,7 +349,7 @@ endif
 
 # threads
 thread_dep = null_dep
-if host_machine.system() != 'windows'
+if host_machine.system() != 'windows' and host_machine.system() != 'emscripten'
   thread_dep = dependency('threads', required: false)
 
   if thread_dep.found()
-- 
2.50.1 (Apple Git-155)

