From 3be6c9919b55f181cfd1140fd1d21a0b08f76832 Mon Sep 17 00:00:00 2001
From: Ian Thomas <ianthomas23@gmail.com>
Date: Mon, 30 Mar 2026 12:54:02 +0100
Subject: [PATCH 1/4] sethostname not static

---
 src/hostname.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/hostname.c b/src/hostname.c
index cd18ef6..3d5c931 100644
--- a/src/hostname.c
+++ b/src/hostname.c
@@ -35,7 +35,7 @@
 #  include <sys/systeminfo.h>
 # endif
 
-static int
+int
 sethostname (char const *name, size_t namelen)
 {
 # if defined HAVE_SYSINFO && defined HAVE_SYS_SYSTEMINFO_H
-- 
2.50.1 (Apple Git-155)

