From 12d3754d9ee4fca12408e01d7d3e0aff9b16b2e6 Mon Sep 17 00:00:00 2001
From: Ian Thomas <ianthomas23@gmail.com>
Date: Mon, 30 Mar 2026 12:55:33 +0100
Subject: [PATCH 2/4] Don't build tests or man pages

---
 Makefile.in | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index ef51f47..752f7d7 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -7246,7 +7246,7 @@ top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 ALL_RECURSIVE_TARGETS = distcheck-hook check-root
-SUBDIRS = po . gnulib-tests
+SUBDIRS = .
 EXTRA_DIST = .mailmap .prev-version .version .vg-suppressions \
 	README-install THANKS.in THANKS-to-translators THANKStt.in \
 	bootstrap bootstrap.conf build-aux/gen-lists-of-programs.sh \
@@ -8735,7 +8735,7 @@ find_upper_case_var = \
 
 @CROSS_COMPILING_TRUE@run_help2man = $(SHELL) $(srcdir)/man/dummy-man
 @BOLD_MAN_REFS_TRUE@@CROSS_COMPILING_FALSE@@HAVE_PERL_TRUE@help2man_OPTS = --bold-refs
-ALL_MANS = $(man1_MANS) $(EXTRA_MANS)
+ALL_MANS = 
 
 # Dependencies common to all man pages.  Updated below.
 
@@ -26919,8 +26919,8 @@ uninstall-am: uninstall-binPROGRAMS uninstall-dvi-am uninstall-html-am \
 
 uninstall-man: uninstall-man1
 
-.MAKE: $(am__recursive_targets) all check check-am install install-am \
-	install-exec install-exec-am install-strip
+.MAKE: $(am__recursive_targets) all check check-am install \
+	install-exec install-strip
 
 .PHONY: $(am__recursive_targets) CTAGS GTAGS TAGS all all-am \
 	am--depfiles am--refresh check check-TESTS check-am \
-- 
2.50.1 (Apple Git-155)

