diff --git a/CMakeLists.txt b/CMakeLists.txt
index f30fd55..3774a2b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,6 +9,13 @@
 cmake_minimum_required(VERSION 3.8)
 project(xcanvas)
 
+if(EMSCRIPTEN)
+    set_property(GLOBAL PROPERTY TARGET_SUPPORTS_SHARED_LIBS TRUE)
+    set(CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS "-s SIDE_MODULE=1")
+    set(CMAKE_SHARED_LIBRARY_CREATE_CXX_FLAGS "-s SIDE_MODULE=1")
+    set(CMAKE_STRIP FALSE)
+endif()
+
 message(STATUS "Forcing tests build type to Release")
 set(CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build." FORCE)
 
@@ -23,6 +30,10 @@ include(CMakePackageConfigHelpers)
 
 set(XCANVAS_INSTALL_LIBRARY_DIR "\"${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}\"")
 
+# Define the path to the shared library for use with CppInterOp dynamic loading
+set(XCANVAS_CPPINTEROP_LIBRARY_PATH "\"/lib/${CMAKE_SHARED_LIBRARY_PREFIX}xcanvas${CMAKE_SHARED_LIBRARY_SUFFIX}\"")
+message(STATUS "XCANVAS_CPPINTEROP_LIBRARY_PATH = ${XCANVAS_CPPINTEROP_LIBRARY_PATH}")
+
 configure_file (
     "${XCANVAS_INCLUDE_DIR}/xcanvas/xcanvas_config_cling.hpp.in"
     "${XCANVAS_INCLUDE_DIR}/xcanvas/xcanvas_config_cling.hpp"
@@ -117,6 +128,7 @@ macro(xcanvas_create_target target_name linkage output_name)
     target_link_libraries(${target_name}
                           PUBLIC xproperty
                           PUBLIC ${XWIDGETS_TARGET_NAME})
+    set_target_properties(${target_name} PROPERTIES NO_SONAME 1)
 
     set_target_properties(${target_name} PROPERTIES
                           PUBLIC_HEADER "${XCANVAS_HEADERS}"
diff --git a/include/xcanvas/xcanvas_config.hpp b/include/xcanvas/xcanvas_config.hpp
index 57f0e43..fbab7f6 100644
--- a/include/xcanvas/xcanvas_config.hpp
+++ b/include/xcanvas/xcanvas_config.hpp
@@ -52,7 +52,7 @@ inline std::string jupyter_canvas_semver()
     return std::string("^") + XJUPYTER_CANVAS_VERSION;
 }
 
-#ifdef __CLING__
+#if defined(__CLANG_REPL__) && defined(__EMSCRIPTEN__)
 #include "xcanvas_config_cling.hpp"
 #endif
 
diff --git a/include/xcanvas/xcanvas_config_cling.hpp.in b/include/xcanvas/xcanvas_config_cling.hpp.in
index 5e1bb04..6897b2b 100644
--- a/include/xcanvas/xcanvas_config_cling.hpp.in
+++ b/include/xcanvas/xcanvas_config_cling.hpp.in
@@ -9,7 +9,15 @@
 #ifndef XCANVAS_CONFIG_CLING_HPP
 #define XCANVAS_CONFIG_CLING_HPP
 
-#pragma cling add_library_path(@XCANVAS_INSTALL_LIBRARY_DIR@)
-#pragma cling load("libxcanvas")
+// #pragma cling add_library_path(@XCANVAS_INSTALL_LIBRARY_DIR@)
+// #pragma cling load("libxcanvas")
+
+#pragma once
+#ifdef __EMSCRIPTEN__
+static bool _xcanvas_loaded = []() {
+    Cpp::LoadLibrary(@XCANVAS_CPPINTEROP_LIBRARY_PATH@, false);
+    return true;
+}();
+#endif
 
 #endif
