From 29479d20326552accb7592b71d8f2d43199cee69 Mon Sep 17 00:00:00 2001
From: Isabel Paredes <isabel.paredes@quantstack.net>
Date: Thu, 6 Mar 2025 11:55:39 +0100
Subject: [PATCH] Disable demangler

---
 inst/include/Rcpp/exceptions_impl.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/inst/include/Rcpp/exceptions_impl.h b/inst/include/Rcpp/exceptions_impl.h
index 4cfecdf..6284d9b 100644
--- a/inst/include/Rcpp/exceptions_impl.h
+++ b/inst/include/Rcpp/exceptions_impl.h
@@ -33,7 +33,8 @@
     defined(_AIX)          || \
     defined(__MUSL__)      || \
     defined(__HAIKU__)     || \
-    defined(__ANDROID__)
+    defined(__ANDROID__)   || \
+    defined(__EMSCRIPTEN__)
 #  define RCPP_DEMANGLER_ENABLED 0
 # elif defined(__GNUC__)  || defined(__clang__)
 #  include <execinfo.h>
