From b0808f37778163d6bdbd97592436a27b3a93a67e Mon Sep 17 00:00:00 2001
From: Isabel Paredes <isabel.paredes@quantstack.net>
Date: Mon, 6 Jan 2025 20:05:25 +0100
Subject: [PATCH] Disable demangler

---
 inst/include/Rcpp/exceptions_impl.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/inst/include/Rcpp/exceptions_impl.h b/inst/include/Rcpp/exceptions_impl.h
index 8f45028..abe0001 100644
--- a/inst/include/Rcpp/exceptions_impl.h
+++ b/inst/include/Rcpp/exceptions_impl.h
@@ -32,7 +32,8 @@
     defined(_AIX)        || \
     defined(__MUSL__)    || \
     defined(__HAIKU__)   || \
-    defined(__ANDROID__)
+    defined(__ANDROID__) || \
+    defined(__EMSCRIPTEN__)
 #  define RCPP_DEMANGLER_ENABLED 0
 # elif defined(__GNUC__)  || defined(__clang__)
 #  include <execinfo.h>
