From bdd92ddece4b8913feb07c1f117122d0acb1d3ea Mon Sep 17 00:00:00 2001
From: Anton Akhmerov <anton.akhmerov@gmail.com>
Date: Tue, 2 Sep 2025 17:30:47 +0200
Subject: [PATCH] disable mumps

---
 setup.py | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/setup.py b/setup.py
index 5bc2a206..28844b38 100755
--- a/setup.py
+++ b/setup.py
@@ -301,16 +301,13 @@ def search_mumps():
         # Conda (via conda-forge).
         ['zmumps_seq', 'mumps_common_seq'],
     ]
-    return next(
-        (libs for libs in lib_sets if all(find_library(lib) for lib in libs)),
-        []
-    )
+    return []
 
 
 def configure_special_extensions(exts, build_summary):
     #### Special config for MUMPS.
     mumps = exts['kwant.linalg._mumps']
-    if 'libraries' in mumps:
+    if False:
         build_summary.append('User-configured MUMPS')
     else:
         mumps_libs = search_mumps()
-- 
2.48.1

