From b5a639ab4dbd0c2d463f1f710c9eada3187c69a9 Mon Sep 17 00:00:00 2001
From: Anton Akhmerov <anton.akhmerov@gmail.com>
Date: Sat, 27 Jul 2024 21:22:45 +0200
Subject: [PATCH] fix scipy compatibility

---
 kwant/physics/gauge.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kwant/physics/gauge.py b/kwant/physics/gauge.py
index eaecdad4..d28a5ed2 100644
--- a/kwant/physics/gauge.py
+++ b/kwant/physics/gauge.py
@@ -754,7 +754,7 @@ def _check_composite_system(syst):
             pre_msg
             + ('There is at least one gauge-fixed loop formed by more than 2 leads. '
                ' The connectivity matrix of the leads is:\n\n'
-               '{}\n\n'.format(metagraph.A))
+               '{}\n\n'.format(metagraph.toarray()))
             + solution_msg
         )
 
-- 
2.45.2

