From 442b296d0da1d554dad1ebf316506144ddaa6e3f Mon Sep 17 00:00:00 2001
From: Fabian Hassler <hassler@physik.rwth-aachen.de>
Date: Wed, 14 May 2025 07:11:16 +0000
Subject: [PATCH] Fix for issue 441

---
 kwant/graph/dijkstra.pyx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kwant/graph/dijkstra.pyx b/kwant/graph/dijkstra.pyx
index 846a04bc..f4754f74 100644
--- a/kwant/graph/dijkstra.pyx
+++ b/kwant/graph/dijkstra.pyx
@@ -46,7 +46,7 @@
 
 import numpy as np
 cimport numpy as np
-from numpy.math cimport INFINITY as inf
+from libc.math cimport INFINITY as inf
 
 from libc.stdlib cimport malloc, free
 from libc.string cimport memset
-- 
2.47.0

