From 8258f74f63ba4f87279c2c02c1181ff9e60ea067 Mon Sep 17 00:00:00 2001
From: Anton Akhmerov <anton.akhmerov@gmail.com>
Date: Sat, 22 Jun 2024 01:09:50 +0200
Subject: [PATCH] wip

---
 kwant/tests/test_kpm.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kwant/tests/test_kpm.py b/kwant/tests/test_kpm.py
index 84ab00fe..1f034112 100644
--- a/kwant/tests/test_kpm.py
+++ b/kwant/tests/test_kpm.py
@@ -35,7 +35,7 @@ p = SimpleNamespace(
 ham = kwant.rmt.gaussian(dim)
 
 # ### Auxiliary functions
-TOL = 1e-12
+TOL = 1e-10
 TOL_SP = 1e-6
 TOL_WEAK = 1e-3
 
-- 
2.45.1

