From d125bf2ff31498f94b922919a232cbfbc3154063 Mon Sep 17 00:00:00 2001
From: Isabel Paredes <isabel.paredes@quantstack.net>
Date: Tue, 17 Feb 2026 18:29:18 +0100
Subject: [PATCH] Fix examples path

---
 pyproject.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index cc21c24..e0bec0d 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -3,7 +3,7 @@ requires = ["maturin>=1.0,<2.0"]
 build-backend = "maturin"
 
 [tool.maturin]
-include = ["./examples/*"]
+include = ["examples/"]
 manifest-path = "lakers-python/Cargo.toml"
 
 [project]
