diff --git a/share/jupyter/kernels/xc11/wasm_kernel.json.in b/share/jupyter/kernels/xc11/wasm_kernel.json.in
index 554591a..a6ea192 100644
--- a/share/jupyter/kernels/xc11/wasm_kernel.json.in
+++ b/share/jupyter/kernels/xc11/wasm_kernel.json.in
@@ -3,9 +3,11 @@
   "argv": [
       "@XEUS_CPP_KERNELSPEC_PATH@xcpp",
       "-resource-dir", "/lib/clang/@CPPINTEROP_LLVM_VERSION_MAJOR@",
+      "-Xclang", "-iwithsysroot/include/compat",
       "-xc",
       "-std=c11",
-      "-mllvm", "-enable-emscripten-sjlj",
+      "-fwasm-exceptions",
+      "-mllvm", "-wasm-enable-sjlj",
       "-msimd128"
   ],
   "language": "c",
diff --git a/share/jupyter/kernels/xc17/wasm_kernel.json.in b/share/jupyter/kernels/xc17/wasm_kernel.json.in
index 2af2c3f..d55a4b5 100644
--- a/share/jupyter/kernels/xc17/wasm_kernel.json.in
+++ b/share/jupyter/kernels/xc17/wasm_kernel.json.in
@@ -3,9 +3,11 @@
   "argv": [
       "@XEUS_CPP_KERNELSPEC_PATH@xcpp",
       "-resource-dir", "/lib/clang/@CPPINTEROP_LLVM_VERSION_MAJOR@",
+      "-Xclang", "-iwithsysroot/include/compat",
       "-xc",
       "-std=c17",
-      "-mllvm", "-enable-emscripten-sjlj",
+      "-fwasm-exceptions",
+      "-mllvm", "-wasm-enable-sjlj",
       "-msimd128"
   ],
   "language": "c",
diff --git a/share/jupyter/kernels/xc23/wasm_kernel.json.in b/share/jupyter/kernels/xc23/wasm_kernel.json.in
index c20a238..c733cd3 100644
--- a/share/jupyter/kernels/xc23/wasm_kernel.json.in
+++ b/share/jupyter/kernels/xc23/wasm_kernel.json.in
@@ -3,9 +3,11 @@
   "argv": [
       "@XEUS_CPP_KERNELSPEC_PATH@xcpp",
       "-resource-dir", "/lib/clang/@CPPINTEROP_LLVM_VERSION_MAJOR@",
+      "-Xclang", "-iwithsysroot/include/compat",
       "-xc",
       "-std=c23",
-      "-mllvm", "-enable-emscripten-sjlj",
+      "-fwasm-exceptions",
+      "-mllvm", "-wasm-enable-sjlj",
       "-msimd128"
   ],
   "language": "c",
diff --git a/share/jupyter/kernels/xcpp17/wasm_kernel.json.in b/share/jupyter/kernels/xcpp17/wasm_kernel.json.in
index fc2a3d6..bcd75de 100644
--- a/share/jupyter/kernels/xcpp17/wasm_kernel.json.in
+++ b/share/jupyter/kernels/xcpp17/wasm_kernel.json.in
@@ -3,8 +3,9 @@
   "argv": [
       "@XEUS_CPP_KERNELSPEC_PATH@xcpp",
       "-resource-dir", "/lib/clang/@CPPINTEROP_LLVM_VERSION_MAJOR@",
-      "-std=c++17", "-mllvm", "-enable-emscripten-cxx-exceptions",
-      "-mllvm", "-enable-emscripten-sjlj", "-msimd128"
+      "-Xclang", "-iwithsysroot/include/compat",
+      "-std=c++17", "-fwasm-exceptions",
+      "-mllvm", "-wasm-enable-sjlj", "-msimd128"
   ],
   "language": "cpp",
   "metadata": {
diff --git a/share/jupyter/kernels/xcpp20/wasm_kernel.json.in b/share/jupyter/kernels/xcpp20/wasm_kernel.json.in
index f84b07d..b7cfd67 100644
--- a/share/jupyter/kernels/xcpp20/wasm_kernel.json.in
+++ b/share/jupyter/kernels/xcpp20/wasm_kernel.json.in
@@ -3,8 +3,9 @@
   "argv": [
       "@XEUS_CPP_KERNELSPEC_PATH@xcpp",
       "-resource-dir", "/lib/clang/@CPPINTEROP_LLVM_VERSION_MAJOR@",
-      "-std=c++20", "-mllvm", "-enable-emscripten-cxx-exceptions",
-      "-mllvm", "-enable-emscripten-sjlj", "-msimd128"
+      "-Xclang", "-iwithsysroot/include/compat",
+      "-std=c++20", "-fwasm-exceptions",
+      "-mllvm", "-wasm-enable-sjlj", "-msimd128"
   ],
   "language": "cpp",
   "metadata": {
diff --git a/share/jupyter/kernels/xcpp23/wasm_kernel.json.in b/share/jupyter/kernels/xcpp23/wasm_kernel.json.in
index 097fd55..960a796 100644
--- a/share/jupyter/kernels/xcpp23/wasm_kernel.json.in
+++ b/share/jupyter/kernels/xcpp23/wasm_kernel.json.in
@@ -3,8 +3,9 @@
   "argv": [
       "@XEUS_CPP_KERNELSPEC_PATH@xcpp",
       "-resource-dir", "/lib/clang/@CPPINTEROP_LLVM_VERSION_MAJOR@",
-      "-std=c++23", "-mllvm", "-enable-emscripten-cxx-exceptions",
-      "-mllvm", "-enable-emscripten-sjlj", "-msimd128"
+      "-Xclang", "-iwithsysroot/include/compat",
+      "-std=c++23", "-fwasm-exceptions",
+      "-mllvm", "-wasm-enable-sjlj", "-msimd128"
   ],
   "language": "cpp",
   "metadata": {
diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 54869f4..bfc065d 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -70,7 +70,7 @@ if(EMSCRIPTEN)
         PUBLIC "SHELL: -fexceptions"
         PUBLIC "SHELL: -s MAIN_MODULE=1"
         PUBLIC "SHELL: -s WASM_BIGINT"
-        PUBLIC "SHELL: -s ASSERTIONS=0"
+        PUBLIC "SHELL: -s ASSERTIONS=1"
         PUBLIC "SHELL: -s ALLOW_MEMORY_GROWTH=1"
         PUBLIC "SHELL: -s STACK_SIZE=32mb"
         PUBLIC "SHELL: -s INITIAL_MEMORY=128mb"
