--- a/cypari/_pari.c
+++ b/cypari/_pari.c
@@ -49436,7 +49436,7 @@
  *         return new_gen(_ret)
  * 
  */
-  __pyx_v__ret = dirpowerssum0(__pyx_v__N, __pyx_v__x, __pyx_v__f, __pyx_v_precision);
+  __pyx_v__ret = dirpowerssum0(__pyx_v__N, __pyx_v__x, __pyx_v__f, 0, __pyx_v_precision);
 
   /* "cypari/auto_instance.pxi":7069
  *         precision = prec_bits_to_words(precision)
@@ -51276,7 +51276,7 @@
  *         return new_gen(_ret)
  * 
  */
-  __pyx_v__ret = ellL1_bitprec(__pyx_v__E, __pyx_v_r, __pyx_v_precision);
+  __pyx_v__ret = ellL1(__pyx_v__E, __pyx_v_r, nbits2prec(__pyx_v_precision));
 
   /* "cypari/auto_instance.pxi":7365
  *             precision = default_bitprec()
@@ -52137,7 +52137,7 @@
  *         return new_gen(_ret)
  * 
  */
-  __pyx_v__ret = ellanalyticrank_bitprec(__pyx_v__E, __pyx_v__eps, __pyx_v_precision);
+  __pyx_v__ret = ellanalyticrank(__pyx_v__E, __pyx_v__eps, nbits2prec(__pyx_v_precision));
 
   /* "cypari/auto_instance.pxi":7482
  *             precision = default_bitprec()
@@ -102325,7 +102325,7 @@
  *         return new_gen(_ret)
  * 
  */
-  __pyx_v__ret = lfunabelianrelinit(__pyx_v__bnfL, __pyx_v__bnfK, __pyx_v__polrel, __pyx_v__sdom, __pyx_v_der, __pyx_v_precision);
+  __pyx_v__ret = (pari_err(e_MISC, "lfunabelianrelinit: PARI 2.17 changed this function's arguments incompatibly"), gen_0);
 
   /* "cypari/auto_instance.pxi":17271
  *             precision = default_bitprec()
@@ -251934,7 +251934,7 @@
  *         return new_gen(_ret)
  * 
  */
-  __pyx_v__ret = dirpowerssum0(__pyx_v__N, __pyx_v__x, __pyx_v__f, __pyx_v_precision);
+  __pyx_v__ret = dirpowerssum0(__pyx_v__N, __pyx_v__x, __pyx_v__f, 0, __pyx_v_precision);
 
   /* "cypari/auto_gen.pxi":6630
  *         precision = prec_bits_to_words(precision)
@@ -253521,7 +253521,7 @@
  *         return new_gen(_ret)
  * 
  */
-  __pyx_v__ret = ellL1_bitprec(__pyx_v__E, __pyx_v_r, __pyx_v_precision);
+  __pyx_v__ret = ellL1(__pyx_v__E, __pyx_v_r, nbits2prec(__pyx_v_precision));
 
   /* "cypari/auto_gen.pxi":6917
  *             precision = default_bitprec()
@@ -254217,7 +254217,7 @@
  *         return new_gen(_ret)
  * 
  */
-  __pyx_v__ret = ellanalyticrank_bitprec(__pyx_v__E, __pyx_v__eps, __pyx_v_precision);
+  __pyx_v__ret = ellanalyticrank(__pyx_v__E, __pyx_v__eps, nbits2prec(__pyx_v_precision));
 
   /* "cypari/auto_gen.pxi":7030
  *             precision = default_bitprec()
@@ -292712,7 +292712,7 @@
  *         return new_gen(_ret)
  * 
  */
-  __pyx_v__ret = lfunabelianrelinit(__pyx_v__bnfL, __pyx_v__bnfK, __pyx_v__polrel, __pyx_v__sdom, __pyx_v_der, __pyx_v_precision);
+  __pyx_v__ret = (pari_err(e_MISC, "lfunabelianrelinit: PARI 2.17 changed this function's arguments incompatibly"), gen_0);
 
   /* "cypari/auto_gen.pxi":15496
  *             precision = default_bitprec()
