From 91a2b3f1aaa8cdcbe3717ba55efdad2e41e4e86e Mon Sep 17 00:00:00 2001
From: Sylvain Corlay <sylvain.corlay@gmail.com>
Date: Sat, 18 Jan 2025 11:03:02 +0100
Subject: [PATCH] Disable threads

---
 meson.build | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/meson.build b/meson.build
index f3e43b595..ed5ee6448 100644
--- a/meson.build
+++ b/meson.build
@@ -366,13 +366,6 @@ endif
 
 # threads
 thread_dep = null_dep
-if host_machine.system() != 'windows'
-  thread_dep = dependency('threads', required: false)
-
-  if thread_dep.found()
-    conf.set('HAVE_PTHREAD', 1)
-  endif
-endif
 
 conf.set_quoted('PACKAGE_NAME', 'HarfBuzz')
 conf.set_quoted('PACKAGE_VERSION', meson.project_version())
-- 
2.39.3 (Apple Git-145)

