From 2a4907c5cacbaa560b28e812906f6b7185cecac5 Mon Sep 17 00:00:00 2001
From: Isabel Paredes <isabel.paredes@quantstack.net>
Date: Fri, 5 Sep 2025 12:59:24 +0200
Subject: [PATCH] Do not include system dirs

---
 setup.py | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/setup.py b/setup.py
index 68945e4..bf4e051 100644
--- a/setup.py
+++ b/setup.py
@@ -199,12 +199,12 @@ class PillowHeifBuildExt(build_ext):
             self._update_extension("_pillow_heif", ["heif"], extra_compile_args=["-O3", "-Werror"])
         else:  # let's assume it's some kind of linux
             # this old code waiting for refactoring, when time comes.
-            self._add_directory(include_dirs, "/usr/local/include")
-            self._add_directory(include_dirs, "/usr/include")
-            self._add_directory(library_dirs, "/usr/local/lib")
-            self._add_directory(library_dirs, "/usr/lib64")
-            self._add_directory(library_dirs, "/usr/lib")
-            self._add_directory(library_dirs, "/lib")
+            # self._add_directory(include_dirs, "/usr/local/include")
+            # self._add_directory(include_dirs, "/usr/include")
+            # self._add_directory(library_dirs, "/usr/local/lib")
+            # self._add_directory(library_dirs, "/usr/lib64")
+            # self._add_directory(library_dirs, "/usr/lib")
+            # self._add_directory(library_dirs, "/lib")
 
             self._update_extension("_pillow_heif", ["heif"], extra_compile_args=["-O3", "-Werror"])
 
-- 
2.51.0

