diff --git a/src/cairo-ps-surface.c b/src/cairo-ps-surface.c
index 381b4cf..0ba2f9f 100644
--- a/src/cairo-ps-surface.c
+++ b/src/cairo-ps-surface.c
@@ -103,11 +103,11 @@
 #endif
 
 #ifndef HAVE_CTIME_R
-static char *ctime_r(const time_t *timep, char *buf)
-{
-    (void)buf;
-    return ctime(timep);
-}
+// static char *ctime_r(const time_t *timep, char *buf)
+// {
+//     (void)buf;
+//     return ctime(timep);
+// }
 #endif
 
 /**
