From 2e183ae5e869bba2ca3e7e1367a513709b3377f6 Mon Sep 17 00:00:00 2001
From: Isabel Paredes <isabel.paredes@quantstack.net>
Date: Tue, 18 Nov 2025 11:40:39 +0100
Subject: [PATCH 1/2] Update linkage

---
 CMakeLists.txt | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2b26284..5bc1a48 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -59,6 +59,7 @@ option(XEUS_R_USE_SHARED_XEUS_R "Link xr  with the xeus-r shared library (instea
 option(XEUS_R_EMSCRIPTEN_WASM_BUILD "Build for wasm with emscripten" OFF)
 
 if(EMSCRIPTEN)
+    set_property(GLOBAL PROPERTY TARGET_SUPPORTS_SHARED_LIBS TRUE)
     add_compile_definitions(XEUS_R_EMSCRIPTEN_WASM_BUILD)
     message("Build with emscripten")
     set(XEUS_R_BUILD_STATIC ON)
@@ -181,7 +182,8 @@ macro(xeus_r_set_common_options target_name)
         CMAKE_CXX_COMPILER_ID MATCHES "Intel")
 
         if (EMSCRIPTEN)
-            target_compile_options(${target_name} PRIVATE -fPIC)
+            target_compile_options(${target_name} PRIVATE -fPIC -fwasm-exceptions)
+            target_link_options(${target_name} PRIVATE -fwasm-exceptions)
         else ()
             target_compile_options(${target_name} PUBLIC -Wunused-parameter -Wextra -Wreorder)
         endif()
@@ -271,8 +273,9 @@ macro(xeus_r_create_target target_name linkage output_name)
     endif ()
 
     if (EMSCRIPTEN)
+        link_directories(${R_HOME}/lib)
         target_link_libraries(${target_name} PUBLIC ${XEUS_R_XEUS_TARGET}
-            PRIVATE ${R_LIBRARY_BLAS} ${R_LIBRARY_LAPACK} ${R_LDFLAGS})
+            PRIVATE ${R_LIBRARY_BASE} ${R_LIBRARY_BLAS} ${R_LIBRARY_LAPACK} embind)
     else ()
         target_link_libraries(${target_name} PUBLIC ${XEUS_R_XEUS_TARGET} PRIVATE ${R_LIBRARY_BASE})
     endif ()
@@ -341,7 +344,6 @@ if(EMSCRIPTEN)
         PUBLIC "SHELL: --pre-js ${CMAKE_CURRENT_SOURCE_DIR}/wasm_patches/env_vars.js"
         PUBLIC "SHELL: --post-js ${CMAKE_CURRENT_SOURCE_DIR}/wasm_patches/post.js"
         PUBLIC "SHELL: -s ERROR_ON_UNDEFINED_SYMBOLS=1"
-        PUBLIC "SHELL: -s EXPORTED_RUNTIME_METHODS='[\"FS\",\"PATH\",\"LDSO\",\"getDylinkMetadata\",\"loadDynamicLibrary\",\"ERRNO_CODES\"]'"
     )
 endif()
 
