--- a/Makefile
+++ b/Makefile
@@ -195,17 +195,10 @@ CJK_CMAPS := source/fitz/cmaps/cns source/fitz/cmaps/gb source/fitz/cmaps/japan
 
 TOFU_SRC := \
 	resources/fonts/urw/Dingbats.cff \
-	resources/fonts/urw/NimbusBoxes-Regular.cff \
-	resources/fonts/urw/NimbusMonoPS-Bold.cff \
-	resources/fonts/urw/NimbusMonoPS-BoldItalic.cff \
-	resources/fonts/urw/NimbusMonoPS-Italic.cff \
-	resources/fonts/urw/NimbusMonoPS-Regular.cff \
-	resources/fonts/urw/NimbusRoman-Bold.cff \
-	resources/fonts/urw/NimbusRoman-BoldItalic.cff \
-	resources/fonts/urw/NimbusRoman-Italic.cff \
-	resources/fonts/urw/NimbusRoman-Regular.cff \
-	resources/fonts/urw/NimbusSans-Bold.cff \
-	resources/fonts/urw/NimbusSans-BoldItalic.cff \
-	resources/fonts/urw/NimbusSans-Italic.cff \
-	resources/fonts/urw/NimbusSans-Regular.cff \
-	resources/fonts/urw/StandardSymbolsPS.cff \
+
+# Disable all font resources for WebAssembly/emscripten builds
+ifeq ($(CC),emcc)
+TOFU_SRC :=
+endif
 
 FONT_BIN := $(TOFU_SRC:%.cff=build/$(OUT)/%.cff.o)