diff --git a/Makefile.riscv64 b/Makefile.riscv64
index 9f6e48b7a..da9848a98 100644
--- a/Makefile.riscv64
+++ b/Makefile.riscv64
@@ -14,7 +14,3 @@ ifeq ($(CORE), RISCV64_ZVL128B)
 CCOMMON_OPT += -march=rv64imafdcv -mabi=lp64d 
 FCOMMON_OPT += -march=rv64imafdcv -mabi=lp64d
 endif
-ifeq ($(CORE), RISCV64_GENERIC)
-CCOMMON_OPT += -march=rv64imafdc -mabi=lp64d
-FCOMMON_OPT += -march=rv64imafdc -mabi=lp64d
-endif
