diff --git a/Makefile.prebuild b/Makefile.prebuild
index b7d695a75..47cbcbf83 100644
--- a/Makefile.prebuild
+++ b/Makefile.prebuild
@@ -71,10 +71,6 @@ ifeq ($(TARGET), RISCV64_ZVL128B)
 TARGET_FLAGS = -march=rv64imafdcv -mabi=lp64d
 endif

-ifeq ($(TARGET), RISCV64_GENERIC)
-TARGET_FLAGS = -march=rv64imafdc -mabi=lp64d
-endif
-
 all: getarch_2nd
	./getarch_2nd  0 >> $(TARGET_MAKE)
	./getarch_2nd  1 >> $(TARGET_CONF)
diff --git a/Makefile.riscv64 b/Makefile.riscv64
index 9f6e48b7a..da9848a98 100644
--- a/Makefile.riscv64
+++ b/Makefile.riscv64
@@ -14,7 +14,3 @@ ifeq ($(CORE), RISCV64_ZVL128B)
 CCOMMON_OPT += -march=rv64imafdcv -mabi=lp64d 
 FCOMMON_OPT += -march=rv64imafdcv -mabi=lp64d
 endif
-ifeq ($(CORE), RISCV64_GENERIC)
-CCOMMON_OPT += -march=rv64imafdc -mabi=lp64d
-FCOMMON_OPT += -march=rv64imafdc -mabi=lp64d
-endif
