From 4803c845b42e02a5fafe3d99c3e9c5f0b039af79 Mon Sep 17 00:00:00 2001
From: Ian Thomas <ianthomas23@gmail.com>
Date: Fri, 13 Mar 2026 09:07:45 +0000
Subject: [PATCH] Clear FEXTRALIB when linking shared library

---
 exports/Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/exports/Makefile b/exports/Makefile
index 9a80fd7ca..47e0aa284 100644
--- a/exports/Makefile
+++ b/exports/Makefile
@@ -88,6 +88,8 @@ ifneq (,$(filter 1 2,$(NOFORTRAN)))
 FEXTRALIB =
 endif
 
+FEXTRALIB =
+
 all::
 
 libs::
-- 
2.51.0

