From 765fdfc97b0fc09ba08c11b698aa25d084f06256 Mon Sep 17 00:00:00 2001
From: Julien Jerphanion <git@jjerphan.xyz>
Date: Thu, 13 Aug 2026 09:45:45 +0200
Subject: Skip fork-based utests for emscripten

Signed-off-by: Julien Jerphanion <git@jjerphan.xyz>
---
 utest/Makefile | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/utest/Makefile b/utest/Makefile
index 07de0c2fc..5b602b1b0 100644
--- a/utest/Makefile
+++ b/utest/Makefile
@@ -41,18 +41,6 @@ endif
 endif
 endif
 
-#this does not work with OpenMP nor with native Windows or Android threads
-# FIXME TBD if this works on OSX, SunOS, POWER and zarch
-ifeq ($(OSNAME), $(filter $(OSNAME),Linux CYGWIN_NT))
-ifneq ($(USE_OPENMP), 1)
-OBJS += test_fork.o
-ifneq ($(NO_LAPACK), 1)
-OBJS += test_post_fork_async.o
-endif
-endif
-OBJS += test_post_fork.o
-endif
-
 ifeq ($(C_COMPILER), PGI)
 OBJS = utest_main2.o
 endif
-- 
2.55.0

