diff --git a/pkgs/sagemath-pari/setup.py b/pkgs/sagemath-pari/setup.py
index a8b04790a0b..88393c13626 100644
--- a/setup.py
+++ b/setup.py
@@ -68,7 +68,7 @@ def run(self):
                "develop":   sage_develop,
                "install":   sage_install,
            },
-           required_modules=('gsl', 'givaro'),
+           required_modules=('flint', 'gsl', 'givaro'),
            optional_modules=('readline',),
            spkgs=['pari', 'gsl', 'givaro'],
            recurse_packages=[
