diff --git a/lib/Interpreter/CppInterOpInterpreter.h b/lib/Interpreter/CppInterOpInterpreter.h
index e358406..760fec5 100644
--- a/lib/Interpreter/CppInterOpInterpreter.h
+++ b/lib/Interpreter/CppInterOpInterpreter.h
@@ -148,8 +148,8 @@ public:
     llvm::InitializeAllAsmPrinters();
 
     std::vector<const char*> vargs(argv + 1, argv + argc);
-    vargs.push_back("-include");
-    vargs.push_back("new");
+    // vargs.push_back("-include");
+    // vargs.push_back("new");
     inner = compat::createClangInterpreter(vargs);
   }
 
