From 213bb16a0f32b932081185837e7275943a499ba3 Mon Sep 17 00:00:00 2001
From: Isabel Paredes <isabel.paredes@quantstack.net>
Date: Tue, 22 Jul 2025 18:22:19 +0200
Subject: [PATCH 03/19] Disable threads

---
 configure | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure b/configure
index 16e8fd3..7ccf66e 100755
--- a/configure
+++ b/configure
@@ -19117,7 +19117,7 @@ return pthread_join ();
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
 then :
-  ax_pthread_ok=yes
+  ax_pthread_ok=no
 fi
 rm -f core conftest.err conftest.$ac_objext conftest.beam \
     conftest$ac_exeext conftest.$ac_ext
@@ -19443,7 +19443,7 @@ pthread_t th; pthread_attr_t attr;
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
 then :
-  ax_pthread_ok=yes
+  ax_pthread_ok=no
 fi
 rm -f core conftest.err conftest.$ac_objext conftest.beam \
     conftest$ac_exeext conftest.$ac_ext
-- 
2.50.1

