From 5e6226845394e9635bd313c4f896c8ea97450242 Mon Sep 17 00:00:00 2001
From: Isabel Paredes <isabel.paredes@quantstack.net>
Date: Tue, 19 Aug 2025 14:24:09 +0200
Subject: [PATCH 07/19] MAYBE Set SIDE_MODULE to 1

---
 configure | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/configure b/configure
index f4ce382..2bb7de2 100755
--- a/configure
+++ b/configure
@@ -27738,8 +27738,8 @@ emscripten*)
   lt_prog_compiler_wl=
   lt_prog_compiler_pic='-fPIC'
   lt_prog_compiler_pic_F77='-fPIC'
-  archive_cmds='$CC -sSIDE_MODULE=2 -shared $libobjs $deplibs $compiler_flags -o $lib'
-  archive_expsym_cmds='$SED "s|^|_|" $export_symbols >$output_objdir/$soname.expsym~$CC -sSIDE_MODULE=2 -shared $libobjs $deplibs $compiler_flags -o $lib -s EXPORTED_FUNCTIONS=@$output_objdir/$soname.expsym'
+  archive_cmds='$CC -sSIDE_MODULE=1 -sWASM_BIGINT=1 -Wl,--allow-multiple-definition $libobjs $deplibs $compiler_flags -o $lib'
+  archive_expsym_cmds='$SED "s|^|_|" $export_symbols >$output_objdir/$soname.expsym~$CC -sSIDE_MODULE=1 -sWASM_BIGINT=1 -Wl,--allow-multiple-definition $libobjs $deplibs $compiler_flags -o $lib -s EXPORTED_FUNCTIONS=@$output_objdir/$soname.expsym'
   archive_cmds_need_lc=no
   no_undefined_flag=
   ;;
@@ -29350,8 +29350,8 @@ fi
             fi'
 	  ;;
   *emcc* | *em++* | *flang* | *flang-new* | *emscripten*)
-    archive_cmds_CXX='$CC -sSIDE_MODULE=2 -shared $libobjs $deplibs $compiler_flags -o $lib'
-    archive_expsym_cmds_CXX='$SED "s|^|_|" $export_symbols >$output_objdir/$soname.expsym~$CC -sSIDE_MODULE=2 -shared $libobjs $deplibs $compiler_flags -o $lib -s EXPORTED_FUNCTIONS=@$output_objdir/$soname.expsym'
+    archive_cmds_CXX='$CC -sSIDE_MODULE=1 -sWASM_BIGINT=1 -Wl,--allow-multiple-definition $libobjs $deplibs $compiler_flags -o $lib'
+    archive_expsym_cmds_CXX='$SED "s|^|_|" $export_symbols >$output_objdir/$soname.expsym~$CC -sSIDE_MODULE=1 -sWASM_BIGINT=1 -Wl,--allow-multiple-definition $libobjs $deplibs $compiler_flags -o $lib -s EXPORTED_FUNCTIONS=@$output_objdir/$soname.expsym'
     hardcode_libdir_flag_spec='-L$libdir'
     hardcode_libdir_flag_spec_CXX='-L$libdir'
     export_dynamic_flag_spec_CXX='$wl--export-all-symbols'
@@ -31929,8 +31929,8 @@ emscripten*)
   lt_prog_compiler_wl_CXX=
   lt_prog_compiler_pic_CXX=='-fPIC'
   lt_prog_compiler_pic_F77='-fPIC'
-  archive_cmds_CXX='$CC -sSIDE_MODULE=2 -shared $libobjs $deplibs $compiler_flags -o $lib'
-  archive_expsym_cmds_CXX='$SED "s|^|_|" $export_symbols >$output_objdir/$soname.expsym~$CC -sSIDE_MODULE=2 -shared $libobjs $deplibs $compiler_flags -o $lib -s EXPORTED_FUNCTIONS=@$output_objdir/$soname.expsym'
+  archive_cmds_CXX='$CC -sSIDE_MODULE=1 -sWASM_BIGINT=1 -Wl,--allow-multiple-definition $libobjs $deplibs $compiler_flags -o $lib'
+  archive_expsym_cmds_CXX='$SED "s|^|_|" $export_symbols >$output_objdir/$soname.expsym~$CC -sSIDE_MODULE=1 -sWASM_BIGINT=1 -Wl,--allow-multiple-definition $libobjs $deplibs $compiler_flags -o $lib -s EXPORTED_FUNCTIONS=@$output_objdir/$soname.expsym'
   archive_cmds_need_lc_CXX=no
   no_undefined_flag_CXX=
   ;;
@@ -35039,8 +35039,8 @@ emscripten*)
   lt_prog_compiler_wl_F77=
   lt_prog_compiler_pic_F77='-fPIC'
   lt_prog_compiler_pic_CXX='-fPIC'
-  archive_cmds_F77='$CC -sSIDE_MODULE=2 -shared $libobjs $deplibs $compiler_flags -o $lib'
-  archive_expsym_cmds_F77='$SED "s|^|_|" $export_symbols >$output_objdir/$soname.expsym~$CC -sSIDE_MODULE=2 -shared $libobjs $deplibs $compiler_flags -o $lib -s EXPORTED_FUNCTIONS=@$output_objdir/$soname.expsym'
+  archive_cmds_F77='$CC -sSIDE_MODULE=1 -sWASM_BIGINT=1 -Wl,--allow-multiple-definition $libobjs $deplibs $compiler_flags -o $lib'
+  archive_expsym_cmds_F77='$SED "s|^|_|" $export_symbols >$output_objdir/$soname.expsym~$CC -sSIDE_MODULE=1 -sWASM_BIGINT=1 -Wl,--allow-multiple-definition $libobjs $deplibs $compiler_flags -o $lib -s EXPORTED_FUNCTIONS=@$output_objdir/$soname.expsym'
   archive_cmds_need_lc_F77=no
   no_undefined_flag_F77=
   ;;
-- 
2.50.1

