From 58aec351233c1fee6d6a1930a5f04652189a6d1c Mon Sep 17 00:00:00 2001
From: Isabel Paredes <isabel.paredes@quantstack.net>
Date: Tue, 19 Aug 2025 14:24:55 +0200
Subject: [PATCH 08/19] Set octave-cli flags

---
 Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index b73da23..3b1cd3f 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -16509,7 +16509,7 @@ src/octave_cli-octave-build-info.$(OBJEXT): src/$(am__dirstamp) \
 
 src/octave-cli$(EXEEXT): $(src_octave_cli_OBJECTS) $(src_octave_cli_DEPENDENCIES) $(EXTRA_src_octave_cli_DEPENDENCIES) src/$(am__dirstamp)
 	@rm -f src/octave-cli$(EXEEXT)
-	$(AM_V_CXXLD)$(src_octave_cli_LINK) $(src_octave_cli_OBJECTS) libmex/.libs/liboctmex.a libinterp/.libs/liboctinterp.a liboctave/.libs/liboctave.a libgnu/.libs/libgnu.a $(LIBS) -lpcre2-8 -lopenblas -lfreetype
+	$(AM_V_CXXLD)$(src_octave_cli_LINK) $(src_octave_cli_OBJECTS) libmex/.libs/liboctmex.a libinterp/.libs/liboctinterp.a liboctave/.libs/liboctave.a libgnu/.libs/libgnu.a $(LIBS)
 src/octave_config-octave-config.$(OBJEXT): src/$(am__dirstamp) \
 	src/$(DEPDIR)/$(am__dirstamp)
 
-- 
2.50.1

