From e70e921f23b57c8fd8f41c5befc337b86729c13d Mon Sep 17 00:00:00 2001
From: Isabel Paredes <isabel.paredes@quantstack.net>
Date: Tue, 19 Aug 2025 18:08:37 +0200
Subject: [PATCH 11/19] Fix duplicate convert_enum error

---
 liboctave/numeric/oct-convn.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/liboctave/numeric/oct-convn.h b/liboctave/numeric/oct-convn.h
index fa339a1..96a4e02 100644
--- a/liboctave/numeric/oct-convn.h
+++ b/liboctave/numeric/oct-convn.h
@@ -138,7 +138,7 @@ extern OCTAVE_API FloatComplexMatrix
 convn (const FloatComplexMatrix& a, const FloatComplexColumnVector& c,
        const FloatComplexRowVector& r, convn_type ct);
 
-convn_type convert_enum (::convn_type ct)
+inline convn_type convert_enum (::convn_type ct)
 {
   switch (ct)
     {
-- 
2.50.1

