From d76c45c06788403eafffff0b6f71363e2cab0cb9 Mon Sep 17 00:00:00 2001
From: Isabel Paredes <isabel.paredes@quantstack.net>
Date: Tue, 19 Aug 2025 18:23:04 +0200
Subject: [PATCH 15/19] Disable multithreading

---
 libinterp/corefcn/gh-manager.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libinterp/corefcn/gh-manager.cc b/libinterp/corefcn/gh-manager.cc
index cfb6260..e4ee392 100644
--- a/libinterp/corefcn/gh-manager.cc
+++ b/libinterp/corefcn/gh-manager.cc
@@ -362,7 +362,7 @@ gh_manager::restore_gcbo ()
 void
 gh_manager::execute_listener (const graphics_handle& h, const octave_value& l)
 {
-  if (octave::thread::is_thread ())
+  if (false) // octave::thread::is_thread ())
     execute_callback (h, l, octave_value ());
   else
     {
-- 
2.50.1

