From 08d7e5a9f7f35224575718d261ba208fcde68760 Mon Sep 17 00:00:00 2001
From: Isabel Paredes <isabel.paredes@quantstack.net>
Date: Thu, 21 Aug 2025 17:24:10 +0200
Subject: [PATCH 18/20] FIXME abort calls native code failure

---
 libinterp/corefcn/interpreter-private.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libinterp/corefcn/interpreter-private.cc b/libinterp/corefcn/interpreter-private.cc
index 0e33ec4..a87254b 100644
--- a/libinterp/corefcn/interpreter-private.cc
+++ b/libinterp/corefcn/interpreter-private.cc
@@ -58,7 +58,7 @@ interpreter& __get_interpreter__ ()
   if (! interp)
     {
       std::cerr << "fatal error: octave interpreter context missing" << std::endl;
-      abort ();
+      // abort ();
     }
 
   return *interp;
