diff --git a/pythran/pythran-linux.cfg b/pythran/pythran-linux.cfg
index a5ea2093d..ddd795b6d 100644
--- a/pythran/pythran-linux.cfg
+++ b/pythran/pythran-linux.cfg
@@ -5,7 +5,7 @@ include_dirs=
 libs=
 library_dirs=
 cflags=-std=c++11 -fno-math-errno -fvisibility=hidden -fno-wrapv -Wno-unused-function -Wno-int-in-bool-context -Wno-unknown-warning-option
-ldflags=-fvisibility=hidden -Wl,-strip-all
+ldflags=-fvisibility=hidden
 blas=blas
 CC=
 CXX=
diff --git a/pythran/pythran-linux2.cfg b/pythran/pythran-linux2.cfg
index a5ea2093d..ddd795b6d 100644
--- a/pythran/pythran-linux2.cfg
+++ b/pythran/pythran-linux2.cfg
@@ -5,7 +5,7 @@ include_dirs=
 libs=
 library_dirs=
 cflags=-std=c++11 -fno-math-errno -fvisibility=hidden -fno-wrapv -Wno-unused-function -Wno-int-in-bool-context -Wno-unknown-warning-option
-ldflags=-fvisibility=hidden -Wl,-strip-all
+ldflags=-fvisibility=hidden
 blas=blas
 CC=
 CXX=
