--- a/configure
+++ b/configure
@@ -14,7 +14,7 @@ PKG_LIBS="-lcairo -lfreetype"
 PKG_TEST_FILE="src/tests/sysdeps.c"

 # Prefer static linking on MacOS
-if [ `uname` = "Darwin" ]; then
+if [ `uname` = "Darwin" ] && [ "${target_platform:-}" != "emscripten-wasm32" ]; then
 PKG_CONFIG_NAME="--static $PKG_CONFIG_NAME"
   if [ `arch` = "i386" ]; then
     export PKG_CONFIG_PATH="$PKG_CONFIG_PATH:/usr/local/opt/libffi/lib/pkgconfig"
@@ -40,7 +40,7 @@ if [ "$INCLUDE_DIR" ] || [ "$LIB_DIR" ]; then
 elif [ "$PKGCONFIG_CFLAGS" ] || [ "$PKGCONFIG_LIBS" ]; then
   echo "Found pkg-config cflags and libs!"
   PKG_CFLAGS=${PKGCONFIG_CFLAGS}
   PKG_LIBS=${PKGCONFIG_LIBS}
-elif [ `uname` = "Darwin" ]; then
+elif [ `uname` = "Darwin" ] && [ "${target_platform:-}" != "emscripten-wasm32" ]; then
   brew --version 2>/dev/null
   if [ $? -eq 0 ]; then
     BREWDIR=`brew --prefix`
