diff --git a/Lib/ctypes/util.py b/Lib/ctypes/util.py
index c550883..d3cda4e 100644
--- a/Lib/ctypes/util.py
+++ b/Lib/ctypes/util.py
@@ -67,6 +67,30 @@ if os.name == "nt":
                 return fname
         return None
 
+elif os.name == "posix" and sys.platform == "emscripten":
+    def _is_wasm(filename):
+        # Return True if the given file is an WASM module
+        wasm_header = b'\x00asm'
+        with open(filename, 'br') as thefile:
+            return thefile.read(4) == wasm_header
+
+    def find_library(name):
+        possible = ['lib%s.so' % name,
+                    'lib%s.wasm' % name]
+
+        paths = os.environ.get('LD_LIBRARY_PATH', '')
+        for dir in paths.split(":"):
+            for name in possible:
+                libfile = os.path.join(dir, name)
+
+                if os.path.isfile(libfile):
+                    if not _is_wasm(libfile):
+                        continue
+
+                    return libfile
+
+        return None
+
 elif os.name == "posix" and sys.platform == "darwin":
     from ctypes.macholib.dyld import dyld_find as _dyld_find
     def find_library(name):
