diff --git a/meson.build b/meson.build
index f58ab5b..be8083f 100644
--- a/meson.build
+++ b/meson.build
@@ -35,9 +35,9 @@ elif host_machine.system() == 'android'
     )
 elif host_machine.system() == 'emscripten'
     plat = 'emscripten'
-    error(
+    warning(
         'The meson buildconfig of pygame-ce does not support emscripten for now. ',
-        'However it may be added in the future',
+        'Experimental emscripten support enabled.',
     )
 else
     # here it one of: cygwin, dragonfly, freebsd, gnu, haiku, netbsd, openbsd, sunos
