diff --git a/CMakeLists.txt b/CMakeLists.txt
index 11f0769..c7e8b92 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -369,7 +369,8 @@ macro(xvega_set_common_options target_name)
             if (CMAKE_SYSTEM_PROCESSOR MATCHES "ppc|powerpc")
                 target_compile_options(${target_name} PUBLIC -mtune=native)
             else ()
-                target_compile_options(${target_name} PUBLIC -march=native)
+                # target_compile_options(${target_name} PUBLIC -march=native)
+                message(STATUS "-march=native is not supported for wasm32-unknown-emscripten")
             endif ()
         endif ()
     endif ()
