diff --git a/setup.py b/setup.py
index 3533226..be93de7 100755
--- a/setup.py
+++ b/setup.py
@@ -188,7 +188,8 @@ def get_ext_modules(use_cython):
                   depends=c_lib[1]['sources'] + c_lib[1]['depends'],
                   include_dirs=[make_ext_path("c"), get_numpy_include()],
                   define_macros=c_macros + cython_macros,
-                  libraries=[c_lib[0]],)
+                  # libraries=[c_lib[0]],
+        )
         for module, source, in zip(cython_modules, cython_sources)
     ]
     return ext_modules
