diff --git a/cpp/cmake_modules/FindSnappyAlt.cmake b/cpp/cmake_modules/FindSnappyAlt.cmake
index 6f84b5cd6..8f59a6c22 100644
--- a/cpp/cmake_modules/FindSnappyAlt.cmake
+++ b/cpp/cmake_modules/FindSnappyAlt.cmake
@@ -54,9 +54,9 @@ if(Snappy_FOUND)
     else()
       # The Conan's Snappy package always uses Snappy::snappy and it's
       # an INTERFACE_LIBRARY.
-      get_target_property(Snappy Snappy::snappy TYPE)
-      if(Snappy_TYPE STREQUAL "STATIC_LIBRARY" OR Snappy_TYPE STREQUAL
-                                                  "INTERFACE_LIBRARY")
+      get_target_property(Snappy_TYPE Snappy::snappy TYPE)
+      if(Snappy_TYPE STREQUAL "STATIC_LIBRARY" OR
+         Snappy_TYPE STREQUAL "INTERFACE_LIBRARY")
         set(Snappy_TARGET Snappy::snappy)
         set(SnappyAlt_FOUND TRUE)
         return()