diff --git a/cpp/cmake_modules/BuildUtils.cmake b/cpp/cmake_modules/BuildUtils.cmake
index 479b20b7a..cc632197c 100644
--- a/cpp/cmake_modules/BuildUtils.cmake
+++ b/cpp/cmake_modules/BuildUtils.cmake
@@ -18,17 +18,8 @@
 # Common path suffixes to be searched by find_library or find_path.
 # Windows artifacts may be found under "<root>/Library", so
 # search there as well.
-set(ARROW_LIBRARY_PATH_SUFFIXES
-    "${CMAKE_LIBRARY_ARCHITECTURE}"
-    "lib/${CMAKE_LIBRARY_ARCHITECTURE}"
-    "lib64"
-    "lib32"
-    "lib"
-    "bin"
-    "Library"
-    "Library/lib"
-    "Library/bin")
-set(ARROW_INCLUDE_PATH_SUFFIXES "include" "Library" "Library/include")
+set(ARROW_LIBRARY_PATH_SUFFIXES "$ENV{PREFIX}/lib")
+set(ARROW_INCLUDE_PATH_SUFFIXES "$ENV{PREFIX}/include")
 
 function(add_thirdparty_lib LIB_NAME LIB_TYPE LIB)
   set(options)
