diff --git a/pyheif/__init__.py b/pyheif/__init__.py
index 108dbdb..fd909bb 100644
--- a/pyheif/__init__.py
+++ b/pyheif/__init__.py
@@ -1,4 +1,4 @@
-import builtins
+# import builtins
 import os
 
 import _libheif_cffi
@@ -7,9 +7,10 @@ from .constants import *
 from .reader import *
 from .writer import *
 
-version_path = os.path.dirname(os.path.abspath(__file__)) + "/data/version.txt"
-with builtins.open(version_path) as f:
-    __version__ = f.read().strip()
+# version_path = os.path.dirname(os.path.abspath(__file__)) + "/data/version.txt"
+# with builtins.open(version_path) as f:
+#     __version__ = f.read().strip()
+__version__ = "0.7.0"
 
 
 def libheif_version():
