diff --git a/libheif/libheif_build.py b/libheif/libheif_build.py
index c33bcb6..c2fa6d2 100644
--- a/libheif/libheif_build.py
+++ b/libheif/libheif_build.py
@@ -10,10 +10,10 @@ with open("libheif/heif.h") as f:
 include_dirs = ["/usr/local/include", "/usr/include", "/opt/local/include"]
 library_dirs = ["/usr/local/lib", "/usr/lib", "/lib", "/opt/local/lib"]
 
-homebrew_prefix = os.getenv("HOMEBREW_PREFIX")
-if homebrew_prefix:
-    include_dirs.append(os.path.join(homebrew_prefix, "include"))
-    library_dirs.append(os.path.join(homebrew_prefix, "lib"))
+build_prefix = os.getenv("PREFIX")
+if build_prefix:
+    include_dirs.append(os.path.join(build_prefix, "include"))
+    library_dirs.append(os.path.join(build_prefix, "lib"))
 
 ffibuilder.set_source(
     "_libheif_cffi",
