From d7b3783019c6c3fe7063f51d8dd96335ef20ddfa Mon Sep 17 00:00:00 2001
From: "duncan.macleod" <duncan.macleod@ligo.org>
Date: Thu, 13 Apr 2023 12:35:03 +0100
Subject: [PATCH 2/5] add macro to fix building on windows

---
 setup.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/setup.py b/setup.py
index ce38467..9e8d292 100644
--- a/setup.py
+++ b/setup.py
@@ -66,6 +66,7 @@ setup(
 			'ligo.segments.__segments',
 			['src/segments.c', 'src/infinity.c', 'src/segment.c', 'src/segmentlist.c'],
 			include_dirs = ['src'],
+			define_macros = [("_CRT_DECLARE_NONSTDC_NAMES", 0)],
 		),
 	],
 	classifiers = [
-- 
2.39.5

