diff --git a/modules/commons/src/cpp/StringHelpers.cpp b/modules/commons/src/cpp/StringHelpers.cpp
index aa52e6150..14e0d288a 100644
--- a/modules/commons/src/cpp/StringHelpers.cpp
+++ b/modules/commons/src/cpp/StringHelpers.cpp
@@ -253,13 +253,7 @@ template <class T>
 bool
 contains(const T& in, const T& needle, size_t npos)
 {
-#ifdef __APPLE__
     return (in.find(needle) != npos);
-#else
-    auto it = std::search(
-        in.begin(), in.end(), std::boyer_moore_searcher(needle.begin(), needle.end()));
-    return (it != in.end());
-#endif
 }
 //=============================================================================
 bool
