diff --git a/modules/engine/src/cpp/CloseAllFiles.cpp b/modules/engine/src/cpp/CloseAllFiles.cpp
index b0178b878..32f208a57 100644
--- a/modules/engine/src/cpp/CloseAllFiles.cpp
+++ b/modules/engine/src/cpp/CloseAllFiles.cpp
@@ -19,23 +19,6 @@ namespace Nelson {
 void
 CloseAllFiles()
 {
-#ifdef _MSC_VER
-    _fcloseall();
-#else
-#if defined(__APPLE__) || defined(__MACH__)
-    FILE* fds_to_close[3]; /* the size being hardcoded to '3' is temporary */
-    int i; /* loop counter */
-    fds_to_close[0] = stdin;
-    fds_to_close[1] = stdout;
-    fds_to_close[2] = stderr;
-    /* max iterations being hardcoded to '3' is temporary: */
-    for ((i = 0); (i < 3); i++) {
-        fclose(fds_to_close[i]);
-    }
-#else
-    fcloseall();
-#endif
-#endif
 }
 //=============================================================================
 } // namespace Nelson
