diff --git a/tclpkg/gv/gv_builtins.c b/tclpkg/gv/gv_builtins.c
index 5c6f43a4..c0a29e1e 100644
--- a/tclpkg/gv/gv_builtins.c
+++ b/tclpkg/gv/gv_builtins.c
@@ -15 +15,8 @@
-lt_symlist_t lt_preloaded_symbols[] = { { 0, 0 } };
+extern gvplugin_library_t gvplugin_dot_layout_LTX_library;
+extern gvplugin_library_t gvplugin_core_LTX_library;
+
+lt_symlist_t lt_preloaded_symbols[] = {
+    { "gvplugin_dot_layout_LTX_library", &gvplugin_dot_layout_LTX_library },
+    { "gvplugin_core_LTX_library", &gvplugin_core_LTX_library },
+    { 0, 0 }
+};
