From f79c6a604fb1a8216246af92ef34641e2402bd45 Mon Sep 17 00:00:00 2001
From: Ian Thomas <ianthomas23@gmail.com>
Date: Fri, 16 Jan 2026 10:10:25 +0000
Subject: [PATCH] Simplify string split code

---
 src/utils/terminal_pager.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/utils/terminal_pager.cpp b/src/utils/terminal_pager.cpp
index e7fe551..1b79996 100644
--- a/src/utils/terminal_pager.cpp
+++ b/src/utils/terminal_pager.cpp
@@ -200,7 +200,7 @@ void terminal_pager::split_input_at_newlines(std::string_view str)
 {
     auto split = str | std::ranges::views::split('\n')
                      | std::ranges::views::transform([](auto&& range) {
-                           return std::string(range.begin(), std::ranges::distance(range));
+                           return std::string(range.begin(), range.end());
                        });
     m_lines = std::vector<std::string>{split.begin(), split.end()};
 }
-- 
2.50.1 (Apple Git-155)

