diff --git a/pkgs/sagemath-singular/setup.py b/pkgs/sagemath-singular/setup.py
index 79ef64b97f3..970f5e51260 100644
--- a/pkgs/sagemath-singular/setup.py
+++ b/pkgs/sagemath-singular/setup.py
@@ -23,7 +23,7 @@
 
 sage_setup(['sagemath-singular'],
            recurse_packages=('sage', 'passagemath_singular'),
-           required_modules=('Singular', 'factory', 'givaro', 'gsl'),
+           required_modules=('flint', 'Singular', 'factory', 'givaro', 'gsl'),
            spkgs=['singular'],
            package_data={"sage": [
                "ext_data/singular/**",
