From ec97f5f19b828539e047b5e255054da7bdb2fbf6 Mon Sep 17 00:00:00 2001
From: Raniere Gaia Costa da Silva <Raniere.CostadaSilva@gesis.org>
Date: Tue, 28 Jul 2026 11:24:49 +0200
Subject: [PATCH] Change function signature

---
 src/tron.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/tron.cpp b/src/tron.cpp
index 35d138d..ac38c3f 100644
--- a/src/tron.cpp
+++ b/src/tron.cpp
@@ -19,10 +19,11 @@ template <class T> static inline T max(T x,T y) { return (x>y)?x:y; }
 extern "C" {
 #endif
 
+// Functions from Netlib's BLAS
 extern double dnrm2_(int *, double *, int *);
 extern double ddot_(int *, double *, int *, double *, int *);
-extern int daxpy_(int *, double *, double *, int *, double *, int *);
-extern int dscal_(int *, double *, double *, int *);
+extern void daxpy_(int *, double *, double *, int *, double *, int *);
+extern void dscal_(int *, double *, double *, int *);
 
 #ifdef __cplusplus
 }
-- 
2.55.0

