diff --git a/CMakeLists.txt b/CMakeLists.txt
index 47a3b1d..c87eb55 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -154,10 +154,16 @@ configure_file(${CMAKE_CURRENT_SOURCE_DIR}/xerces-c.pc.in
   DESTINATION "${PKGCONFIGDIR}"
   COMPONENT "development")
 
+option(XERCES_BUILD_DOCS "Build Xerces documentation targets" ON)
+option(XERCES_BUILD_SAMPLES "Build Xerces sample programs" ON)
+option(XERCES_BUILD_TESTS "Build Xerces test programs" ON)
+
 # Process subdirectories
-add_subdirectory(doc)
-add_subdirectory(src)
-add_subdirectory(tests)
-add_subdirectory(samples)
+if(XERCES_BUILD_DOCS)
+  add_subdirectory(doc)
+endif()
+add_subdirectory(src)
+if(XERCES_BUILD_TESTS)
+  add_subdirectory(tests)
+endif()
+if(XERCES_BUILD_SAMPLES)
+  add_subdirectory(samples)
+endif()
 
 # Display configuration summary
