--- a/configure
+++ b/configure
@@ -20,11 +20,11 @@
 # Building paths
-PATH=".:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/bin:/sbin:$PATH"
-CPATH=".:/usr/local/include:$CPATH"
-LIBRARY_PATH=".:/usr/local/lib:$LIBRARY_PATH"
+PATH=".:$PATH:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/bin:/sbin"
+CPATH=".:$CPATH:/usr/local/include"
+LIBRARY_PATH=".:$LIBRARY_PATH:/usr/local/lib"
 if test -z "$LD_LIBRARY_PATH" -a -n "$DYLD_LIBRARY_PATH"
 then
   LD_LIBRARY_PATH="$DYLD_LIBRARY_PATH"
 fi
-LD_LIBRARY_PATH=".:/usr/local/lib:$LD_LIBRARY_PATH"
-PKG_CONFIG_PATH=".:/usr/local/lib/pkgconfig:$PKG_CONFIG_PATH"
+LD_LIBRARY_PATH=".:$LD_LIBRARY_PATH:/usr/local/lib"
+PKG_CONFIG_PATH=".:$PKG_CONFIG_PATH:/usr/local/lib/pkgconfig"
 export PATH CPATH LIBRARY_PATH LD_LIBRARY_PATH PKG_CONFIG_PATH
