diff --git a/setup.py b/setup.py
index 401a050..96e738f 100644
--- a/setup.py
+++ b/setup.py
@@ -39,8 +39,8 @@ class NumpyExtension(setuptools.Extension):
 
     @property
     def include_dirs(self):
-        from numpy import get_include
-        return self._include_dirs + [get_include()]
+        # from numpy import get_include
+        return self._include_dirs #+ [get_include()]
 
     @include_dirs.setter
     def include_dirs(self, include_dirs):
