From 0bd7e4432d6e28d8168b67a775f5a1152a5ce27b Mon Sep 17 00:00:00 2001
From: Sylvain Corlay <sylvain.corlay@gmail.com>
Date: Sat, 18 Jan 2025 10:32:53 +0100
Subject: [PATCH] Disable pthreads

---
 meson.build | 22 ----------------------
 1 file changed, 22 deletions(-)

diff --git a/meson.build b/meson.build
index 162b2927c..3c9e4a2d1 100644
--- a/meson.build
+++ b/meson.build
@@ -709,28 +709,6 @@ endforeach
 
 conf.set('HAVE_STRNDUP', cc.has_function('strndup', prefix : '#include <string.h>'))
 
-pthread_c_args = []
-pthread_link_args = []
-
-foreach thread_flags : check_thread_flags
-  if not conf.has('CAIRO_HAS_PTHREAD')
-    cflags = thread_flags[0]
-    lflags = thread_flags[1]
-    real_pthread = thread_flags.length() > 2 ? thread_flags[2].get('real', true) : true
-
-    if cc.links(files('meson-cc-tests/pthread.c'), args: cflags + lflags, name: 'pthreads')
-      conf.set('CAIRO_HAS_PTHREAD', 1)
-      if real_pthread
-        conf.set('CAIRO_HAS_REAL_PTHREAD', 1)
-      endif
-      pthread_c_args = cflags
-      pthread_link_args = lflags
-    endif
-  endif
-endforeach
-
-extra_link_args += pthread_link_args
-
 # Atomics are an optional feature in C11. Also need to check that C11 atomics are lock free.
 # Windows can't use C11 atomics as some files are compiled with C++.
 if host_machine.system() != 'windows'
-- 
2.39.3 (Apple Git-145)

