diff --git a/dolo/algos/invert.py b/dolo/algos/invert.py
--- a/dolo/algos/invert.py
+++ b/dolo/algos/invert.py
@@ -0,0 +1,2 @@
+import sys
+
@@ -157 +159 @@
-target = "parallel"
+target = "cpu" if sys.platform == "emscripten" else "parallel"
diff --git a/dolo/numeric/serial_operations.py b/dolo/numeric/serial_operations.py
--- a/dolo/numeric/serial_operations.py
+++ b/dolo/numeric/serial_operations.py
@@ -1,0 +2 @@
+import sys
@@ -109,0 +111,2 @@
+target = "cpu" if sys.platform == "emscripten" else "parallel"
+
@@ -111 +115 @@
-    "void(f8[:,:], f8[:])", "(m,n)->(m)", target="parallel"
+    "void(f8[:,:], f8[:])", "(m,n)->(m)", target=target
