From e909b3b6487eb58e074bb500ef92a2c1aeb8e83d Mon Sep 17 00:00:00 2001
From: Sylvain Corlay <sylvain.corlay@gmail.com>
Date: Sat, 18 Jan 2025 10:49:07 +0100
Subject: [PATCH] Disable threads

---
 meson.build | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/meson.build b/meson.build
index 4439758..4094484 100644
--- a/meson.build
+++ b/meson.build
@@ -296,10 +296,6 @@ else
   conf.set('FLEXIBLE_ARRAY_MEMBER', 1)
 endif
 
-if cc.compiles(files('meson-cc-tests/pthread-prio-inherit-test.c'))
-  conf.set('HAVE_PTHREAD_PRIO_INHERIT', 1)
-endif
-
 if cc.links(files('meson-cc-tests/stdatomic-primitives-test.c'), name: 'stdatomic.h atomics')
   conf.set('HAVE_STDATOMIC_PRIMITIVES', 1)
 endif
@@ -381,12 +377,6 @@ if fc_cachedir in ['yes', 'no', 'default']
   endif
 endif
 
-if host_machine.system() != 'windows'
-  thread_dep = dependency('threads')
-  conf.set('HAVE_PTHREAD', 1)
-  deps += [thread_dep]
-endif
-
 fc_templatedir = get_option('template-dir')
 if fc_templatedir in ['default', 'yes', 'no']
   fc_templatedir = prefix / get_option('datadir') / 'fontconfig/conf.avail'
-- 
2.39.3 (Apple Git-145)

