diff --git a/meson.build b/meson.build
index 08d9532..fae3333 100644
--- a/meson.build
+++ b/meson.build
@@ -289,12 +289,6 @@ if fc_cachedir in ['yes', 'no', 'default']
   endif
 endif
 
-if host_machine.system() != 'windows'
-  thread_dep = dependency('threads')
-  conf.set('HAVE_PTHREAD', 1)
-  deps += [thread_dep]
-endif
-
 fc_templatedir = get_option('template-dir')
 if fc_templatedir in ['default', 'yes', 'no']
   fc_templatedir = prefix / get_option('datadir') / 'fontconfig/conf.avail'
