diff --git a/pkgs/sagemath-linbox/setup.py b/pkgs/sagemath-linbox/setup.py
index 53b2c1ed16f..a7b39823ff1 100644
--- a/setup.py
+++ b/setup.py
@@ -9,7 +9,7 @@
 
 sage_setup(['sagemath-linbox'],
            recurse_packages=('sage', 'passagemath_linbox'),
-           required_modules=('fflas-ffpack', 'givaro', 'gsl', 'linbox', 'cblas',
+           required_modules=('fflas-ffpack', 'flint', 'givaro', 'gsl', 'linbox', 'cblas',
                              'zlib'),
            spkgs=['linbox', 'gsl'],
            package_data={})
