--- freetype/raw.py
+++ freetype/raw.py
@@ -11,5 +11,6 @@
 import os
 import platform
+import glob
 from ctypes import *
 import ctypes.util
 
@@ -29,4 +30,9 @@
 else:
     library_name = 'libfreetype.so'
 
+if system == 'Emscripten':
+    libraries = sorted(glob.glob('/lib/libfreetype.so*'))
+    if libraries:
+        library_name = libraries[0]
+
 filename = os.path.join(os.path.dirname(freetype.__file__), library_name)
