diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2a6a8238a..32f4c704c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -96,7 +96,7 @@ else()
   if(CMAKE_SYSTEM_NAME MATCHES "PICO")
     add_compile_options(-c -Wall -Wextra -Wno-unused -Wno-strict-prototypes -pipe -g -O0)
   elseif(UNIX)
-    add_compile_options(-c -Wall -Wextra -Werror -Wshadow -Wunused -Wstrict-prototypes -pipe -g -O0)
+    add_compile_options(-c -Wall -Wextra -Wno-missing-braces -Wshadow -Wunused -Wstrict-prototypes -pipe -g -O0)
     # C99 pedantic doesn't like struct anonymous in unix header
     if (NOT CMAKE_C_STANDARD STREQUAL "99")
       add_compile_options(-Wpedantic)
