diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2a6a8238a..b0a694474 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -87,11 +87,8 @@ string(TOUPPER ${CMAKE_BUILD_TYPE} CMAKE_BUILD_TYPE)
 
 # Compile options
 if(CMAKE_BUILD_TYPE MATCHES "RELEASE" OR "Release")
-  if(UNIX)
-    add_compile_options(-pipe -O3)
-  elseif(CMAKE_SYSTEM_NAME MATCHES "Generic")
-    add_compile_options(-pipe -O3)
-  endif()
+  message(WARNING "TEST THAT THIS IS HAPPENING")
+  add_compile_options(-pipe -O3 -Wno-error -Wno-missing-braces -Wno-missing-field-initializers -Wno-missing-prototypes)
 else()
   if(CMAKE_SYSTEM_NAME MATCHES "PICO")
     add_compile_options(-c -Wall -Wextra -Wno-unused -Wno-strict-prototypes -pipe -g -O0)
