diff --git a/tclpkg/gv/gv_builtins.c b/tclpkg/gv/gv_builtins.c
index 5c6f43a4..c0a29e1e 100644
--- a/tclpkg/gv/gv_builtins.c
+++ b/tclpkg/gv/gv_builtins.c
@@ -15 +15,8 @@
-lt_symlist_t lt_preloaded_symbols[] = { { 0, 0 } };
+extern gvplugin_library_t gvplugin_dot_layout_LTX_library;
+extern gvplugin_library_t gvplugin_core_LTX_library;
+
+lt_symlist_t lt_preloaded_symbols[] = {
+    { "gvplugin_dot_layout_LTX_library", &gvplugin_dot_layout_LTX_library },
+    { "gvplugin_core_LTX_library", &gvplugin_core_LTX_library },
+    { 0, 0 }
+};
diff --git a/tclpkg/gv/CMakeLists.txt b/tclpkg/gv/CMakeLists.txt
--- a/tclpkg/gv/CMakeLists.txt
+++ b/tclpkg/gv/CMakeLists.txt
@@ -296,6 +296,10 @@
     target_link_libraries(gv_python3
       PRIVATE cdt cgraph gvc ${Python3_LIBRARIES}
     )
+    if(NOT BUILD_SHARED_LIBS)
+      target_link_libraries(gv_python3
+        PRIVATE gvplugin_core gvplugin_dot_layout)
+    endif()
 
     install(
       TARGETS gv_python3
