diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -794,8 +794,8 @@
 endif()
 
 if(NOT ENABLE_PYTHON STREQUAL "OFF")
-  find_package(Python3 COMPONENTS Development)
-  if(ENABLE_SWIG AND SWIG_python_FOUND AND Python3_Development_FOUND)
+  find_package(Python3 COMPONENTS Development.Module)
+  if(ENABLE_SWIG AND SWIG_python_FOUND AND Python3_Development.Module_FOUND)
     if(ENABLE_PYTHON STREQUAL "AUTO")
       message(STATUS "setting -DENABLE_PYTHON=ON")
       set(ENABLE_PYTHON ON)
diff --git a/tclpkg/gv/CMakeLists.txt b/tclpkg/gv/CMakeLists.txt
--- a/tclpkg/gv/CMakeLists.txt
+++ b/tclpkg/gv/CMakeLists.txt
@@ -296,7 +296,7 @@
       PROPERTY SWIG_GENERATED_COMPILE_OPTIONS "${PYTHON_FLAGS}"
     )
     target_link_libraries(gv_python3
-      PRIVATE cdt cgraph gvc ${Python3_LIBRARIES}
+      PRIVATE cdt cgraph gvc Python3::Module
     )
     if(NOT BUILD_SHARED_LIBS)
       target_link_libraries(gv_python3
