From ae0c9fcc591c30fa69d18a2241048020ef3c5bbd Mon Sep 17 00:00:00 2001
From: Mick Batchelor <mickbatch98@gmail.com>
Date: Thu, 10 Aug 2023 00:12:22 +1000
Subject: [PATCH 2/2] fix wasm extension for dmake

---
 lib/smallvector.h    | 15 ++-------------
 tools/CMakeLists.txt | 16 +++++++++++++---
 2 files changed, 15 insertions(+), 16 deletions(-)

diff --git a/lib/smallvector.h b/lib/smallvector.h
index 42c45a173..00f2df7f1 100644
--- a/lib/smallvector.h
+++ b/lib/smallvector.h
@@ -32,25 +32,14 @@ using SmallVector = boost::container::small_vector<T, N>;
 #include <utility>
 #include <vector>
 
-template<class T, std::size_t N>
-struct TaggedAllocator : std::allocator<T>
-{
-    template<class ... Ts>
-    // cppcheck-suppress noExplicitConstructor
-    // NOLINTNEXTLINE(google-explicit-constructor)
-    TaggedAllocator(Ts&&... ts)
-        : std::allocator<T>(std::forward<Ts>(ts)...)
-    {}
-};
-
 template<typename T, std::size_t N = DefaultSmallVectorSize>
-class SmallVector : public std::vector<T, TaggedAllocator<T, N>>
+class SmallVector : public std::vector<T>
 {
 public:
     template<class ... Ts>
     // NOLINTNEXTLINE(google-explicit-constructor)
     SmallVector(Ts&&... ts)
-        : std::vector<T, TaggedAllocator<T, N>>(std::forward<Ts>(ts)...)
+        : std::vector<T>(std::forward<Ts>(ts)...)
     {
         this->reserve(N);
     }
diff --git a/tools/CMakeLists.txt b/tools/CMakeLists.txt
index cbf3f8bcc..02ba65e62 100644
--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -9,6 +9,10 @@ foreach(file ${srcs_lib})
     set(srcs_tools ${srcs_tools} ${src})
 endforeach()
 
+if (EMSCRIPTEN)
+    set(CMAKE_EXECUTABLE_SUFFIX ".wasm")
+endif()
+
 add_executable(dmake EXCLUDE_FROM_ALL
         dmake.cpp
         ${CMAKE_SOURCE_DIR}/cli/filelister.cpp
@@ -25,6 +29,12 @@ if (WIN32 AND NOT BORLAND)
     endif()
 endif()
 
-add_custom_target(run-dmake $<TARGET_FILE:dmake>
-        WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-        DEPENDS dmake)
+if (NOT EMSCRIPTEN)
+    add_custom_target(run-dmake $<TARGET_FILE:dmake>
+            WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+            DEPENDS dmake)
+endif()
+
+if (EMSCRIPTEN)
+    set(CMAKE_EXECUTABLE_SUFFIX ".wasm")
+endif()
-- 
2.39.2.windows.1

