From 2959d8befd389ca435b13353db7265920b345507 Mon Sep 17 00:00:00 2001
From: Isabel Paredes <isabel.paredes@quantstack.net>
Date: Tue, 15 Apr 2025 13:36:33 +0200
Subject: [PATCH] Disable faer-sparce

---
 Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index de1d924..f168664 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -57,7 +57,7 @@ julia = ["sdp", "dep:libc", "dep:num-derive",  "serde", "faer-sparse", "buildinf
 # This is the behaviour used when building the python wheels
 # for distribution on pypi
 
-python = ["sdp", "dep:libc", "dep:pyo3", "dep:num-derive", "serde", "faer-sparse", "buildinfo"]
+python = ["sdp", "dep:libc", "dep:pyo3", "dep:num-derive", "serde", "buildinfo"]
 
 #compile with faer supernodal solver option
 faer-sparse = ["dep:faer", "dep:faer-traits"]
