diff --git a/Cargo.toml b/Cargo.toml
index 741628c..f36c664 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -53,7 +53,7 @@ julia = ["sdp", "dep:libc", "dep:num-derive",  "serde", "faer-sparse"]
 # NB: python builds use scipy shared libraries
 # for blas/lapack, and should *not* explicitly 
 # enable a blas/lapack source package 
-python = ["sdp", "dep:libc", "dep:pyo3", "dep:num-derive", "serde", "faer-sparse"]
+python = ["sdp", "dep:libc", "dep:pyo3", "dep:num-derive", "serde"]
 
 wasm = ["dep:web-time"]
 
