From ec4164b71152d510d6c5ee6d116f54aed04874b9 Mon Sep 17 00:00:00 2001
From: AntoinePrv <AntoinePrv@users.noreply.github.com>
Date: Thu, 3 Apr 2025 12:37:58 +0200
Subject: [PATCH] Force detection of shared libs

---
 configure | 16 ++--------------
 1 file changed, 2 insertions(+), 14 deletions(-)

diff --git a/configure b/configure
index 629c461..789f0b7 100755
--- a/configure
+++ b/configure
@@ -38121,20 +38121,8 @@ ac_compiler_gnu=$ac_cv_c_compiler_gnu
 
 # Only expand once:
 
-
-
-if test $enable_shared = yes; then
-  SHARED_LIBS=yes
-else
-  SHARED_LIBS=no
-  # Note: Octave doesn't build currently without shared libraries.  And it
-  #       might not work or make sense at all to build static Octave libraries.
-  #       All existing logic for this use-case is still in place in case
-  #       someone would like to base on that to build static versions of the
-  #       libraries.
-  as_fn_error $? "Building shared libraries is required!" "$LINENO" 5
-fi
-
+enable_shared=yes
+SHARED_LIBS=yes
 
 if test $enable_static = yes; then
   STATIC_LIBS=yes
-- 

