From c471b9ce6f726101563f7738844e4dbb7bd6fd7b Mon Sep 17 00:00:00 2001
From: Sam Gardner <sam@wx4stg.com>
Date: Sun, 16 Mar 2025 16:32:35 -0500
Subject: [PATCH] dont use hdf5 parallel if available but not requested

---
 configure | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/configure b/configure
index ae48feca8d..7d7f45f85f 100755
--- a/configure
+++ b/configure
@@ -27628,7 +27628,9 @@ printf "%s\n" "#define HDF5_HAS_COLL_METADATA_OPS 1" >>confdefs.h
 
    # If parallel is available in hdf5, enable it in the C code. Also add some stuff to netcdf.h.
    if test "x$ac_cv_func_H5Pget_fapl_mpio" = xyes -o "x$ac_cv_func_H5Pget_fapl_mpiposix" = xyes; then
-      hdf5_parallel=yes
+      if test "x$enable_parallel" = xyes; then
+          hdf5_parallel=yes
+      fi
    fi
    { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking whether parallel io is enabled in hdf5" >&5
 printf %s "checking whether parallel io is enabled in hdf5... " >&6; }
