diff --git a/CMakeLists.txt b/CMakeLists.txt
index 02d2f1e..4d498b0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -182,11 +182,7 @@ foreach(fbs_file IN LISTS FLATBUFFERS_SCHEMAS)
     list(APPEND FLATBUFFERS_GENERATED_HEADERS "${FLATBUFFERS_GENERATED_DIR}/${header_name}_generated.h")
 endforeach()
 
-if(TARGET flatbuffers::flatc)
-    set(FLATC_EXECUTABLE flatbuffers::flatc)
-else()
-    set(FLATC_EXECUTABLE flatc)
-endif()
+set(FLATC_EXECUTABLE flatc)
 
 add_custom_command(
     OUTPUT ${FLATBUFFERS_GENERATED_HEADERS}
