--- a/src/lib/libdylink.js	2025-05-15 18:06:42
+++ b/src/lib/libdylink.js	2025-09-16 08:59:41
@@ -461,7 +461,7 @@
       failIf(name !== 'dylink.0');
     }
 
-    var customSection = { neededDynlibs: [], tlsExports: new Set(), weakImports: new Set(), runtimePaths: [] };
+    var customSection = { neededDynlibs: [], tlsExports: new Set(), weakImports: new Set(), runtimePaths: ['/lib'] };
     var WASM_DYLINK_MEM_INFO = 0x1;
     var WASM_DYLINK_NEEDED = 0x2;
     var WASM_DYLINK_EXPORT_INFO = 0x3;
