From e69736a1e0611d838d15d9e03bbed95b67f4797a Mon Sep 17 00:00:00 2001
From: martinRenou <martin.renou@gmail.com>
Date: Tue, 16 Dec 2025 13:57:31 +0100
Subject: [PATCH 1/1] Do not fail when failing to resolve symbols

---
 src/lib/libdylink.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lib/libdylink.js b/src/lib/libdylink.js
index fdfe2e608..7c2e487ee 100644
--- a/src/lib/libdylink.js
+++ b/src/lib/libdylink.js
@@ -327,7 +327,7 @@ var LibraryDylink = {
           entry.value = value;
 #endif
         } else {
-          throw new Error(`bad export type for '${symName}': ${typeof value}`);
+          console.warn(`bad export type for '${symName}': ${typeof value}: This can potentially be ignored (some shared library has not yet been loaded but will eventually be)`);
         }
       }
     }
--
2.52.0

