diff --git a/CMakeLists.txt b/CMakeLists.txt
index bb1b832..eae64b5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,6 +3,11 @@ cmake_minimum_required(VERSION 3.24...3.29)
 project(BOOST_HISTOGRAM LANGUAGES CXX)
 # Version is added later
 
+set_property(GLOBAL PROPERTY TARGET_SUPPORTS_SHARED_LIBS TRUE)
+set(CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS "-s SIDE_MODULE=1")
+set(CMAKE_SHARED_LIBRARY_CREATE_CXX_FLAGS "-s SIDE_MODULE=1")
+set(CMAKE_STRIP FALSE)  # used by default in pybind11 on .so modules
+
 include(FetchContent)
 
 # Boost histogram requires C++14
