diff --git a/llvm/cmake/modules/AddLLVM.cmake b/llvm/cmake/modules/AddLLVM.cmake
index b92c244d628d..1144605c411e 100644
--- a/llvm/cmake/modules/AddLLVM.cmake
+++ b/llvm/cmake/modules/AddLLVM.cmake
@@ -2490,6 +2490,7 @@ function(llvm_add_tool_symlink project link_name target)
 
   if (${target} IN_LIST LLVM_DRIVER_TOOLS)
     set_property(GLOBAL APPEND PROPERTY LLVM_DRIVER_TOOL_ALIASES_${target} ${link_name})
+    return()
   endif()
   set(dest_binary "$<TARGET_FILE:${target}>")
 
