From f6dd31e45d291a477c3e79f518d84ce128a486ad Mon Sep 17 00:00:00 2001
From: Matthias Koeppe <mkoeppe@math.ucdavis.edu>
Date: Fri, 17 Jul 2026 14:27:40 -0700
Subject: [PATCH] src/ulong_extras/is_square.c: Make mod{63,64,65} static

---
 src/ulong_extras/is_square.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/ulong_extras/is_square.c b/src/ulong_extras/is_square.c
index 8ffe61964..322bde868 100644
--- a/src/ulong_extras/is_square.c
+++ b/src/ulong_extras/is_square.c
@@ -12,15 +12,15 @@
 #include <math.h>
 #include "ulong_extras.h"
 
-int mod64[64] = {1,1,0,0,1,0,0,0,0,1,0,0,0,0,0,0,1,1,0,0,0,
+static int mod64[64] = {1,1,0,0,1,0,0,0,0,1,0,0,0,0,0,0,1,1,0,0,0,
                  0,0,0,0,1,0,0,0,0,0,0,0,1,0,0,1,0,0,0,0,1,
                  0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,1,0,0,0,0,0,0};
 
-int mod65[65] = {1,1,0,0,1,0,0,0,0,1,1,0,0,0,1,0,1,0,0,0,0,0,
+static int mod65[65] = {1,1,0,0,1,0,0,0,0,1,1,0,0,0,1,0,1,0,0,0,0,0,
                  0,0,0,1,1,0,0,1,1,0,0,0,0,1,1,0,0,1,1,0,0,0,
                  0,0,0,0,0,1,0,1,0,0,0,1,1,0,0,0,0,1,0,0,1};
 
-int mod63[63] = {1,1,0,0,1,0,0,1,0,1,0,0,0,0,1,0,1,0,1,0,0,
+static int mod63[63] = {1,1,0,0,1,0,0,1,0,1,0,0,0,0,1,0,1,0,1,0,0,
                  0,1,0,0,1,0,0,1,0,0,0,0,0,0,1,1,1,0,0,0,0,
                  0,1,0,0,1,0,0,1,0,0,0,0,0,0,1,0,1,0,0,0,0};
 
-- 
2.51.1

