From 8461b17ae1bbefa77de0c52697f190b871381656 Mon Sep 17 00:00:00 2001
From: Isabel Paredes <isabel.paredes@quantstack.net>
Date: Fri, 3 Jul 2026 17:30:19 +0200
Subject: [PATCH] Build shared lib with emcc

---
 src/Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile.in b/src/Makefile.in
index 260e1ad..38758aa 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -71,7 +71,7 @@ OBJECTS = init.o mapclip.o mapget.o smooth.o thin.o
 libdir = ../libs
 
 shlib: objects
-	"$(R_HOME)/bin/R" CMD SHLIB -o $(SHLIB) $(OBJECTS)
+	$(CC) $(SHLIB_LDFLAGS_R) -o $(SHLIB) $(OBJECTS)
 
 objects:
 	$(MAKE) -f "$(R_HOME)/etc$(R_ARCH)/Makeconf" -f Makefile $(OBJECTS)
