From ed18218029bf9450205a2fee26a601b700b79588 Mon Sep 17 00:00:00 2001
From: Isabel Paredes <isabel.paredes@quantstack.net>
Date: Mon, 8 Dec 2025 07:51:01 +0000
Subject: [PATCH 3/8] Add emscripten as supported arch

---
 getarch.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/getarch.c b/getarch.c
index b51c3ed64..b1493ee9f 100644
--- a/getarch.c
+++ b/getarch.c
@@ -1895,6 +1895,11 @@ USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #define OPENBLAS_SUPPORTED
 #endif
 
+#ifdef __EMSCRIPTEN__
+#include "cpuid_riscv64.c"
+#define OPENBLAS_SUPPORTED
+#endif
+
 #ifndef OPENBLAS_SUPPORTED
 #error "This arch/CPU is not supported by OpenBLAS."
 #endif
-- 
2.51.0

