diff --git a/image/decode/segdec.c b/image/decode/segdec.c
index fb83f2b..1eb9ae4 100644
--- a/image/decode/segdec.c
+++ b/image/decode/segdec.c
@@ -52,6 +52,9 @@ static Int DecodeSignificantAbsLevel (struct CAdaptiveHuffman *pAHexpt, BitIOInf
 //================================================================
 // Memory access functions
 //================================================================
+
+U32 _byteswap_ulong(U32 bits);
+
 static U32 _FORCEINLINE _load4(void* pv)
 {