From 2f3ea873fb9be6f98d13c02f2a93da077c01d05f Mon Sep 17 00:00:00 2001
From: Raniere Gaia Costa da Silva <Raniere.CostadaSilva@gesis.org>
Date: Wed, 1 Apr 2026 10:16:43 +0200
Subject: [PATCH 3/3] Set UNAME to Emscripten

---
 src/Makevars | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makevars b/src/Makevars
index 017d5d2c..660285e0 100644
--- a/src/Makevars
+++ b/src/Makevars
@@ -1,4 +1,4 @@
-UNAME := $(shell uname)
+UNAME := Emscripten
 
 LIBUV := libuv-1.44.2
 
-- 
2.52.0

