diff --git a/CMakeLists.txt b/CMakeLists.txt
index e758b5dc8..4d7f5150a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -254,7 +254,7 @@ check_library_exists(m pow "" HAVE_LIBM)
 #-----------------------------------------------------------------------------
 # Target geos: C++ API library
 #-----------------------------------------------------------------------------
-add_library(geos "")
+add_library(geos STATIC "")
 add_library(GEOS::geos ALIAS geos)
 target_link_libraries(geos PUBLIC geos_cxx_flags PRIVATE $<BUILD_INTERFACE:ryu>)
 # ryu is an object library, nothing is actually being linked here. The BUILD_INTERFACE
