From a9b50756f95a2a2736680c2ce7609efec7fe77be Mon Sep 17 00:00:00 2001
From: Ian Thomas <ianthomas23@gmail.com>
Date: Wed, 4 Mar 2026 13:04:18 +0000
Subject: [PATCH 1/2] Add cmake shared library flags

---
 CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2576d31..c7013ca 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,6 +25,11 @@ set(matio_LIB_VERSIONINFO "14:0:0")
 
 set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 
+set_property(GLOBAL PROPERTY TARGET_SUPPORTS_SHARED_LIBS TRUE)
+set(CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS "-s SIDE_MODULE=1")
+set(CMAKE_SHARED_LIBRARY_CREATE_CXX_FLAGS "-s SIDE_MODULE=1")
+set(CMAKE_STRIP FALSE)  # used by default in pybind11 on .so modules
+
 # Installation options
 include(GNUInstallDirs)
 
-- 
2.51.0

