From 354cd55b416ecc39db2c255fa7917fd9a8be62e1 Mon Sep 17 00:00:00 2001
From: Ian Thomas <ianthomas23@gmail.com>
Date: Wed, 4 Mar 2026 13:04:56 +0000
Subject: [PATCH 2/2] Do not build matdump exe

---
 CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c7013ca..197a628 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -40,5 +40,4 @@ include(cmake/staticAnalyzers.cmake)
 include(cmake/compilerOptions.cmake)
 include(cmake/getopt.cmake)
 include(cmake/src.cmake)
-include(cmake/tools.cmake)
 include(cmake/test.cmake)
-- 
2.51.0

